<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Belinda Cornes" />

<meta name="date" content="2021-07-08" />

<title>Step 1: Sample QC [Redone Samples]</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.5.1/htmlwidgets.js"></script>
<script src="site_libs/d3-4.9.0/d3.min.js"></script>
<link href="site_libs/colorbrewer-1.0.0/colorbrewer.css" rel="stylesheet" />
<script src="site_libs/colorbrewer-1.0.0/colorbrewer.js"></script>
<script src="site_libs/d3-tip-0.7.1/index-min.js"></script>
<link href="site_libs/d3panels-1.4.9/d3panels.min.css" rel="stylesheet" />
<script src="site_libs/d3panels-1.4.9/d3panels.min.js"></script>
<script src="site_libs/qtlcharts_iplot-0.11.6/iplot.js"></script>
<script src="site_libs/iplot-binding-0.11.6/iplot.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<script src="site_libs/qtlcharts_iboxplot-0.11.6/iboxplot.js"></script>
<script src="site_libs/iboxplot-binding-0.11.6/iboxplot.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Korstanje-OConnor Workflow</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheJacksonLaboratory/Korstanje-OConnor_Workflow">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Step 1: Sample QC [Redone Samples]</h1>
<h4 class="author">Belinda Cornes</h4>
<h4 class="date">2021-07-08</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#loading-project">Loading Project</a></li>
<li><a href="#missing-data">Missing Data</a></li>
<li><a href="#sex">Sex</a></li>
<li><a href="#sample-duplicates">Sample Duplicates</a></li>
<li><a href="#array-intensities">Array Intensities</a></li>
<li><a href="#genotype-frequencies">Genotype Frequencies</a></li>
<li><a href="#crossover-counts">Crossover Counts</a></li>
<li><a href="#genotyping-error-lod-scores">Genotyping Error LOD Scores</a></li>
<li><a href="#apparent-genotyping-errors">Apparent Genotyping Errors</a></li>
<li><a href="#removing-samples">Removing Samples</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-07-08
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>Korstanje-OConnor_Workflow/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div class="panel-group" id="workflowr-checks">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210317code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210317)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210317code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210317)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/corneb/Documents/MyJax/CS/Projects/Korstanje-Oconnor/workflowr/Korstanje-OConnor_Workflow
</td>
<td style="text-align:left;">
.
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheJacksonLaboratoryKorstanjeOConnorWorkflowtree59ea5436f1fd34e5c2826c5850747155ca08165btargetblank59ea543a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/TheJacksonLaboratory/Korstanje-OConnor_Workflow/tree/59ea5436f1fd34e5c2826c5850747155ca08165b" target="_blank">59ea543</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomTheJacksonLaboratoryKorstanjeOConnorWorkflowtree59ea5436f1fd34e5c2826c5850747155ca08165btargetblank59ea543a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/TheJacksonLaboratory/Korstanje-OConnor_Workflow/tree/59ea5436f1fd34e5c2826c5850747155ca08165b" target="_blank">59ea543</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  data/bad_markers_all_3batches.RData
    Untracked:  data/bad_markers_all_wcont_3batches.RData
    Untracked:  data/bad_markers_all_wcont_redone.RData
    Untracked:  data/e_3batches.RData
    Untracked:  data/e_g_snpg_samqc_3batches.RData
    Untracked:  data/e_g_snpg_samqc_wcont_3batches.RData
    Untracked:  data/e_g_snpg_samqc_wcont_redone.RData
    Untracked:  data/e_wcont_3batches.RData
    Untracked:  data/e_wcont_redone.RData
    Untracked:  data/errors_ind_3batches.RData
    Untracked:  data/errors_ind_wcont_3batches.RData
    Untracked:  data/errors_ind_wcont_redone.RData
    Untracked:  data/genotype_errors_marker_3batches.RData
    Untracked:  data/genotype_errors_marker_wcont_3batches.RData
    Untracked:  data/genotype_errors_marker_wcont_redone.RData
    Untracked:  data/genotype_freq_marker_3batches.RData
    Untracked:  data/genotype_freq_marker_wcont_3batches.RData
    Untracked:  data/genotype_freq_marker_wcont_redone.RData
    Untracked:  data/gm_allqc_3batches.RData
    Untracked:  data/gm_allqc_wcont_3batches.RData
    Untracked:  data/gm_allqc_wcont_redone.RData
    Untracked:  data/gm_samqc_3batches.RData
    Untracked:  data/gm_samqc_wcont_3batches.RData
    Untracked:  data/gm_samqc_wcont_redone.RData
    Untracked:  data/m_3batches.RData
    Untracked:  data/m_wcont_3batches.RData
    Untracked:  data/m_wcont_redone.RData
    Untracked:  data/nxo_3batches.RData
    Untracked:  data/nxo_wcont_3batches.RData
    Untracked:  data/nxo_wcont_redone.RData
    Untracked:  data/percent_missing_id_3batches.RData
    Untracked:  data/percent_missing_id_wcont_3batches.RData
    Untracked:  data/percent_missing_id_wcont_redone.RData
    Untracked:  data/percent_missing_marker_3batches.RData
    Untracked:  data/percent_missing_marker_wcont_3batches.RData
    Untracked:  data/percent_missing_marker_wcont_redone.RData
    Untracked:  data/qc_info_bad_sample_3batches.RData
    Untracked:  data/qc_info_bad_sample_wcont_3batches.RData
    Untracked:  data/qc_info_bad_sample_wcont_redone.RData
    Untracked:  data/snpg_3batches.RData
    Untracked:  data/snpg_wcont_3batches.RData
    Untracked:  data/snpg_wcont_redone.RData
    Untracked:  data/summary.cg_3batches.RData
    Untracked:  data/summary.cg_wcont_3batches.RData
    Untracked:  data/summary.cg_wcont_redone.RData
    Untracked:  output/Percent_genotype_errors_obs_vs_predicted_3batches.pdf
    Untracked:  output/Percent_genotype_errors_obs_vs_predicted_wcont_3batches.pdf
    Untracked:  output/Percent_genotype_errors_obs_vs_predicted_wcont_redone.pdf
    Untracked:  output/Percent_missing_genotype_data_3batches.pdf
    Untracked:  output/Percent_missing_genotype_data_per_marker_3batches.pdf
    Untracked:  output/Percent_missing_genotype_data_per_marker_wcont_3batches.pdf
    Untracked:  output/Percent_missing_genotype_data_per_marker_wcont_redone.pdf
    Untracked:  output/Percent_missing_genotype_data_wcont_3batches.pdf
    Untracked:  output/Percent_missing_genotype_data_wcont_redone.pdf
    Untracked:  output/genotype_error_marker_3batches.pdf
    Untracked:  output/genotype_error_marker_wcont_3batches.pdf
    Untracked:  output/genotype_error_marker_wcont_redone.pdf
    Untracked:  output/genotype_frequency_marker_3batches.pdf
    Untracked:  output/genotype_frequency_marker_wcont_3batches.pdf
    Untracked:  output/genotype_frequency_marker_wcont_redone.pdf
    Untracked:  output/number_crossover_3batches.pdf
    Untracked:  output/number_crossover_wcont_3batches.pdf
    Untracked:  output/number_crossover_wcont_redone.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/2.1_sample_bqc_wcont_redone.Rmd</code>) and HTML (<code>docs/2.1_sample_bqc_wcont_redone.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheJacksonLaboratory/Korstanje-OConnor_Workflow/blob/59ea5436f1fd34e5c2826c5850747155ca08165b/analysis/2.1_sample_bqc_wcont_redone.Rmd" target="_blank">59ea543</a>
</td>
<td>
Belinda Cornes
</td>
<td>
2021-07-08
</td>
<td>
redone samples
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This script is running genotype QC on raw data (with some outcomes already seen in the <a href="summary.html">project at a glace</a>). Here, we first load the R/qtl2 package and the data. We’ll also load the R/broman package for some utilities and plotting functions, and R/qtlcharts for interactive graphs.</p>
<p>We will follow the steps by <code>Karl Broman</code> found <a href="https://kbroman.org/qtl2/assets/vignettes/do_diagnostics.html">here</a></p>
<div id="loading-project" class="section level3">
<h3>Loading Project</h3>
<pre class="r"><code>gm &lt;- read_cross2(&quot;/Users/corneb/Documents/MyJax/CS/Projects/Korstanje-Oconnor/haplotype.reconstruction/output/qtl2/gm.json&quot;)

gm</code></pre>
<pre><code>Object of class cross2 (crosstype &quot;do&quot;)

Total individuals                43
No. genotyped individuals        43
No. phenotyped individuals       43
No. with both geno &amp; pheno       43

No. phenotypes                    1
No. covariates                    3
No. phenotype covariates          0

No. chromosomes                  20
Total markers                137220

No. markers by chr:
    1     2     3     4     5     6     7     8     9    10    11    12    13 
10415 10432  8203  7951  8028  8125  7756  6714  6979  6628  7430  6441  6323 
   14    15    16    17    18    19     X 
 6227  5531  5174  5320  4784  3676  5083 </code></pre>
<pre class="r"><code>sample_file &lt;- dir(path = filepaths, pattern = &quot;^DODB_*&quot;, full.names = TRUE)
samples &lt;- read.csv(sample_file)
all.equal(as.character(ind_ids(gm)), as.character(samples$Unique.Sample.ID))</code></pre>
<pre><code>[1] TRUE</code></pre>
</div>
<div id="missing-data" class="section level3">
<h3>Missing Data</h3>
<pre class="r"><code>percent_missing &lt;- n_missing(gm, &quot;ind&quot;, &quot;prop&quot;)*100

labels &lt;- paste0(as.character(do.call(rbind.data.frame, strsplit(names(percent_missing), &quot;_&quot;))[,6]), &quot; (&quot;, round(percent_missing,2), &quot;%)&quot;)
iplot(seq_along(percent_missing), percent_missing, indID=labels,
      chartOpts=list(xlab=&quot;Mouse&quot;, ylab=&quot;Percent missing genotype data&quot;,
                     ylim=c(0, 70)))</code></pre>
<pre><code>Set screen size to height=700 x width=1000</code></pre>
<div id="htmlwidget-f3b64798a4a1b8f30209" style="width:912px;height:576px;" class="iplot html-widget"></div>
<script type="application/json" data-for="htmlwidget-f3b64798a4a1b8f30209">{"x":{"data":{"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43],"y":[4.4338,3.5979,3.5738,3.49,32.796,4.3696,4.3113,3.9717,4.1466,3.6518,3.8063,3.7866,7.9201,7.5062,4.2188,3.5549,3.4638,4.5431,3.5286,4.9271,3.7633,5.4926,4.0701,3.9404,14.98,5.3323,5.4402,4.6633,6.0319,4.6954,6.7315,24.462,10.73,8.7276,5.0474,6.2061,36.828,34.092,5.1815,5.2434,3.8478,5.1815,4.8783],"group":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"indID":["X-5187-20 (4.43%)","X-5467-20 (3.6%)","X-8542-20 (3.57%)","X-8638-20 (3.49%)","X-5190-20 (32.8%)","X-5490-20 (4.37%)","X-8544-20 (4.31%)","X-5195-20 (3.97%)","X-5499-20 (4.15%)","X-8560-20 (3.65%)","X-5198-20 (3.81%)","X-5507-20 (3.79%)","X-8572-20 (7.92%)","X-5232-20 (7.51%)","X-5509-20 (4.22%)","X-8575-20 (3.55%)","X-5237-20 (3.46%)","X-5548-20 (4.54%)","X-8577-20 (3.53%)","X-2173-20 (4.93%)","X-5244-20 (3.76%)","X-8294-20 (5.49%)","X-8578-20 (4.07%)","X-2178-20 (3.94%)","X-5443-20 (14.98%)","X-8296-20 (5.33%)","X-8585-20 (5.44%)","X-2198-20 (4.66%)","X-5448-20 (6.03%)","X-8303-20 (4.7%)","X-8586-20 (6.73%)","X-5150-20 (24.46%)","X-5455-20 (10.73%)","X-8306-20 (8.73%)","X-8610-20 (5.05%)","X-5169-20 (6.21%)","B6-1622 (36.83%)","X-8503-20 (34.09%)","X-8614-20 (5.18%)","X-5181-20 (5.24%)","X-5466-20 (3.85%)","X-8528-20 (5.18%)","X-8637-20 (4.88%)"]},"chartOpts":{"xlab":"Mouse","ylab":"Percent missing genotype data","ylim":[0,70]}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>#save into pdf
pdf(file = &quot;output/Percent_missing_genotype_data_wcont_redone.pdf&quot;, width = 20, height = 20)
#labels &lt;- as.character(do.call(rbind.data.frame, strsplit(names(totxo), &quot;V01_&quot;))[,2])
labels &lt;- as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), &quot;_&quot;))[,6])
#labels &lt;- paste0(names(percent_missing), &quot; (&quot;, round(percent_missing,2), &quot;%)&quot;)
labels[percent_missing &lt; 10] = &quot;&quot;
# Change point shapes and colors
p &lt;- ggplot(data = data.frame(Mouse=seq_along(percent_missing),  
                         Percent_missing_genotype_data = percent_missing,
                         batch = factor(as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), &quot;_&quot;))[,5]))
                         #batch = factor(as.character(do.call(rbind.data.frame, strsplit(as.character(samples$Directory), &quot;_&quot;))[,5]))
                         ), 
        aes(x=Mouse, y=Percent_missing_genotype_data, color = batch)) +
        geom_point() +
        geom_hline(yintercept=10, linetype=&quot;solid&quot;, color = &quot;red&quot;) +
        geom_text_repel(aes(label=labels), vjust = 0, nudge_y = 0.01, show.legend = FALSE, size=3) +
        theme(text = element_text(size = 10))
p

dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>p</code></pre>
<p><img src="figure/2.1_sample_bqc_wcont_redone.Rmd/Missing%20data%20per%20sample-2.png" width="912" style="display: block; margin: auto;" /></p>
<pre class="r"><code>save(percent_missing,file = &quot;data/percent_missing_id_wcont_redone.RData&quot;)

gm.covar = data.frame(id=rownames(gm$covar),gm$covar)
qc_info_cr &lt;- merge(gm.covar,
                  data.frame(id = names(percent_missing),percent_missing = percent_missing,stringsAsFactors = F),by = &quot;id&quot;)
bad.sample.cr &lt;- qc_info_cr[qc_info_cr$percent_missing &gt;= 10,]</code></pre>
<p>There are 6 mice that are missing &gt;10% of their genotypes. These samples are:</p>
<table class="table table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;">
Neogen_Sample_ID
</th>
<th style="text-align:center;">
Genoprobs_Sample_ID
</th>
<th style="text-align:center;">
percent_missing
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;width: 5cm; ">
B6-1622
</td>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210604_B6-1622_B12
</td>
<td style="text-align:center;width: 5cm; ">
36.8277219064276
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cm; ">
X-5150-20
</td>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-5150-20_E11
</td>
<td style="text-align:center;width: 5cm; ">
24.4621775251421
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cm; ">
X-5443-20
</td>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-5443-20_F10
</td>
<td style="text-align:center;width: 5cm; ">
14.979594811252
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cm; ">
X-5455-20
</td>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-5455-20_F11
</td>
<td style="text-align:center;width: 5cm; ">
10.7302142544819
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cm; ">
X-8503-20
</td>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-8503-20_C12
</td>
<td style="text-align:center;width: 5cm; ">
34.0919691007142
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cm; ">
X-5190-20
</td>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210701_X-5190-20_H5
</td>
<td style="text-align:center;width: 5cm; ">
32.7962396152164
</td>
</tr>
</tbody>
</table>
</div>
<div id="sex" class="section level3">
<h3>Sex</h3>
<pre class="r"><code>hdf5_filename &lt;- dir(path = filepaths, pattern = &quot;^hdf5_*&quot;, full.names = TRUE)
snps_file &lt;- &quot;/Users/corneb/Documents/MyJax/CS/Projects/QTL2_support.files/gm_uwisc_v1.csv&quot;
snps &lt;- read.csv(snps_file)

snps &lt;- snps[snps$unique == TRUE, ]
#snps &lt;- snps[snps$chr %in% c(1:19, &quot;X&quot;), ]
snps$chr &lt;- sub(&quot;^chr&quot;, &quot;&quot;, snps$chr)  ###remove prefix &quot;chr&quot;
colnames(snps)[colnames(snps)==&quot;bp_mm10&quot;] &lt;- &quot;pos&quot; 
colnames(snps)[colnames(snps)==&quot;cM_cox&quot;] &lt;- &quot;cM&quot;
snps &lt;- snps %&gt;% drop_na(chr, marker) 
snps$pos &lt;- snps$pos * 1e-6
rownames(snps) &lt;- snps$marker
colnames(snps)[1:4] &lt;- c(&quot;marker&quot;, &quot;chr&quot;, &quot;pos&quot;, &quot;pos&quot;) 

#  g &lt;- h5read(hdf5_filename, &quot;G&quot;)
#  g &lt;- do.call(cbind, g)
x &lt;- h5read(hdf5_filename, &quot;X&quot;) # X channel intensities
x &lt;- do.call(cbind, x)
y &lt;- h5read(hdf5_filename, &quot;Y&quot;) # Y channel intensities
y &lt;- do.call(cbind, y)
rn &lt;- h5read(hdf5_filename, &quot;rownames&quot;)[[1]]  # markers 
cn &lt;- h5read(hdf5_filename, &quot;colnames&quot;)  # samples
cn &lt;- do.call(c, cn)
# dimnames(g) &lt;- list(rn, cn)
dimnames(x) &lt;- list(rn, cn)
dimnames(y) &lt;- list(rn, cn)
#cr &lt;- colMeans(g != &quot;--&quot;) # Call rate for each sample avg 0.95
#  sex &lt;- determine_sex(x = x, y = y, markers = snps)$se

markers &lt;- snps

chrx &lt;- markers$marker[which(markers$chr == &quot;X&quot;)]
chry &lt;- markers$marker[which(markers$chr == &quot;Y&quot;)]
#x[chrx,ind_ids(gm)]

chrx_int &lt;- colMeans(x[chrx,as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), &quot;_&quot;))[,6])] + y[chrx,as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), &quot;_&quot;))[,6])], na.rm = T)
chry_int &lt;- colMeans(x[chry,as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), &quot;_&quot;))[,6])] + y[chry,as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), &quot;_&quot;))[,6])], na.rm = T)
#  df &lt;- data.frame(x = chrx_int, y = chry_int)
#  mc &lt;- Mclust(data = df, G = 2, model = &quot;VVV&quot;, prior=priorControl()) 
#  # Determine which cluster is females by selecting the cluster with the
#  # higher Chr X intensity mean.
#  cluster_means &lt;- mc$parameters$mean
#  female_cl &lt;- which.max(cluster_means[&quot;x&quot;,])
#  sex &lt;- rep(&quot;M&quot;, length(mc$classification))
#  names(sex) &lt;- colnames(x)
#  sex[mc$classification == female_cl] &lt;- &quot;F&quot;
#  return(list(sex = sex))

#xint &lt;- read_csv_numer(&quot;/Users/corneb/Documents/MyJax/CS/Projects/Simon.John.Tolman/Data/GM/Simon.John.Tolman_qtl2_chrXint.csv&quot;, transpose=TRUE)
#yint &lt;- read_csv_numer(&quot;data/Univ_of_Penn_Thaiss_qtl2_chrYint.csv&quot;, transpose=TRUE)

#xint2 &lt;- t(x[chrx,ind_ids(gm)])
#yint2 &lt;- t(y[chry,ind_ids(gm)])

#sample_file &lt;- dir(path = filepaths, pattern = &quot;^DODB_*&quot;, full.names = TRUE)
#samples &lt;- read.csv(sample_file)

all.equal(as.character(ind_ids(gm)), as.character(samples$Unique.Sample.ID))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>#sex order
sex &lt;- samples$Sex
#sex &lt;- gm$covar[rownames(xint),&quot;sexp&quot;]
#sex &lt;- substr(rownames(xint), 1, 1)

#x_pval &lt;- apply(xint2, 2, function(a) t.test(a ~ sex)$p.value)
#y_pval &lt;- apply(yint2, 2, function(a) t.test(a ~ sex)$p.value)

#xint_ave &lt;- rowMeans(xint2[, x_pval &lt; 0.05/length(x_pval)], na.rm=TRUE)
#yint_ave &lt;- rowMeans(yint2[, y_pval &lt; 0.05/length(y_pval)], na.rm=TRUE)

#BAF = yint / (xint + yint)
#LRR = log2((xint + yint) / (xint_ave+yint_ave))

point_colors &lt;- as.character( brocolors(&quot;web&quot;)[c(&quot;red&quot;,&quot;green&quot;, &quot;purple&quot;)] )
percent_missing &lt;- n_missing(gm, summary=&quot;proportion&quot;)*100
labels &lt;- paste0(names(chrx_int), &quot; (&quot;, round(percent_missing), &quot;%)&quot;)
iplot( chrx_int,  chry_int, group=sex, indID=labels,
      chartOpts=list(pointcolor=point_colors, pointsize=4,
                     xlab=&quot;Average X chr intensity&quot;, ylab=&quot;Average Y chr intensity&quot;))</code></pre>
<div id="htmlwidget-9c52c1f69abe72785df3" style="width:912px;height:576px;" class="iplot html-widget"></div>
<script type="application/json" data-for="htmlwidget-9c52c1f69abe72785df3">{"x":{"data":{"x":[0.88526,1.1212,1.0865,0.8749,1.0734,0.83593,1.1518,0.86813,0.85333,1.1268,0.8728,0.84702,0.93647,1.1789,0.86486,0.87611,1.1045,0.9004,0.85466,0.83367,1.0631,1.0587,0.85141,1.0973,1.1534,1.0518,0.82771,1.0348,1.0546,1.0529,0.83246,1.1878,1.0617,0.83906,1.0493,1.0416,1.2865,1.7882,1.0332,1.1023,1.0569,0.82142,0.80416],"y":[0.65458,0.06729,0.070032,0.7291,0.57029,0.62448,0.070645,0.69313,0.64129,0.040452,0.69529,0.68439,0.68674,0.066839,0.67035,0.71548,0.051138,0.59316,0.71148,0.63019,0.032903,0.13535,0.70416,0.056871,0.064935,0.093613,0.63981,0.083839,0.096129,0.080452,0.63152,0.20026,0.096258,0.61881,0.090839,0.079774,0.73765,1.0232,0.087032,0.071613,0.045667,0.60526,0.59087],"group":[3,2,2,3,3,3,2,3,3,2,3,3,3,2,3,3,2,3,3,3,2,2,3,2,2,2,3,2,2,2,3,2,2,3,2,2,1,3,2,2,2,3,3],"indID":["X-5187-20 (4%)","X-5467-20 (4%)","X-8542-20 (4%)","X-8638-20 (3%)","X-5190-20 (33%)","X-5490-20 (4%)","X-8544-20 (4%)","X-5195-20 (4%)","X-5499-20 (4%)","X-8560-20 (4%)","X-5198-20 (4%)","X-5507-20 (4%)","X-8572-20 (8%)","X-5232-20 (8%)","X-5509-20 (4%)","X-8575-20 (4%)","X-5237-20 (3%)","X-5548-20 (5%)","X-8577-20 (4%)","X-2173-20 (5%)","X-5244-20 (4%)","X-8294-20 (5%)","X-8578-20 (4%)","X-2178-20 (4%)","X-5443-20 (15%)","X-8296-20 (5%)","X-8585-20 (5%)","X-2198-20 (5%)","X-5448-20 (6%)","X-8303-20 (5%)","X-8586-20 (7%)","X-5150-20 (24%)","X-5455-20 (11%)","X-8306-20 (9%)","X-8610-20 (5%)","X-5169-20 (6%)","B6-1622 (37%)","X-8503-20 (34%)","X-8614-20 (5%)","X-5181-20 (5%)","X-5466-20 (4%)","X-8528-20 (5%)","X-8637-20 (5%)"]},"chartOpts":{"pointcolor":["#ff4136","#2ecc40","#b10dc9"],"pointsize":4,"xlab":"Average X chr intensity","ylab":"Average Y chr intensity"}},"evals":[],"jsHooks":[]}</script>
<p>For figures above and below, those labelled as female in metadata given, are coloured <code>green</code>, with those labelled as male are coloured as <code>purple</code>. The above is an interactive scatterplot of the average SNP intensity on the Y chromosome versus the average SNP intensity on the X chromosome.</p>
<p>The cluster in the upper-left are male mice (low X chromosome intensity and high Y chromosome intensity), and the cluster in the lower-right are female mice (high X chromosome intensity and low Y chromsome intensity).</p>
<pre class="r"><code>phetX &lt;- rowSums(gm$geno$X == 2)/rowSums(gm$geno$X != 0)
phetX &lt;- phetX[as.character(do.call(rbind.data.frame, strsplit(names(phetX), &quot;_&quot;))[,6]) %in% names(chrx_int)]
names(phetX) &lt;- as.character(do.call(rbind.data.frame, strsplit(names(phetX), &quot;_&quot;))[,6])
iplot(chrx_int, phetX, group=sex, indID=labels,
      chartOpts=list(pointcolor=point_colors, pointsize=4,
                     xlab=&quot;Average X chr intensity&quot;, ylab=&quot;Proportion het on X chr&quot;))</code></pre>
<div id="htmlwidget-4415da58db86afdaf3b4" style="width:912px;height:576px;" class="iplot html-widget"></div>
<script type="application/json" data-for="htmlwidget-4415da58db86afdaf3b4">{"x":{"data":{"x":[0.88526,1.1212,1.0865,0.8749,1.0734,0.83593,1.1518,0.86813,0.85333,1.1268,0.8728,0.84702,0.93647,1.1789,0.86486,0.87611,1.1045,0.9004,0.85466,0.83367,1.0631,1.0587,0.85141,1.0973,1.1534,1.0518,0.82771,1.0348,1.0546,1.0529,0.83246,1.1878,1.0617,0.83906,1.0493,1.0416,1.2865,1.7882,1.0332,1.1023,1.0569,0.82142,0.80416],"y":[0.018896,0.23767,0.20275,0.012661,0.28571,0.018652,0.16628,0.011081,0.012775,0.27009,0.0098203,0.012091,0.03979,0.32391,0.017511,0.010795,0.22556,0.019001,0.011229,0.031076,0.18597,0.2838,0.015253,0.2729,0.29584,0.13924,0.03532,0.23861,0.22372,0.12476,0.050905,0.24583,0.23411,0.077803,0.29579,0.27706,0.027696,0.043094,0.16772,0.12219,0.1375,0.028292,0.024448],"group":[3,2,2,3,3,3,2,3,3,2,3,3,3,2,3,3,2,3,3,3,2,2,3,2,2,2,3,2,2,2,3,2,2,3,2,2,1,3,2,2,2,3,3],"indID":["X-5187-20 (4%)","X-5467-20 (4%)","X-8542-20 (4%)","X-8638-20 (3%)","X-5190-20 (33%)","X-5490-20 (4%)","X-8544-20 (4%)","X-5195-20 (4%)","X-5499-20 (4%)","X-8560-20 (4%)","X-5198-20 (4%)","X-5507-20 (4%)","X-8572-20 (8%)","X-5232-20 (8%)","X-5509-20 (4%)","X-8575-20 (4%)","X-5237-20 (3%)","X-5548-20 (5%)","X-8577-20 (4%)","X-2173-20 (5%)","X-5244-20 (4%)","X-8294-20 (5%)","X-8578-20 (4%)","X-2178-20 (4%)","X-5443-20 (15%)","X-8296-20 (5%)","X-8585-20 (5%)","X-2198-20 (5%)","X-5448-20 (6%)","X-8303-20 (5%)","X-8586-20 (7%)","X-5150-20 (24%)","X-5455-20 (11%)","X-8306-20 (9%)","X-8610-20 (5%)","X-5169-20 (6%)","B6-1622 (37%)","X-8503-20 (34%)","X-8614-20 (5%)","X-5181-20 (5%)","X-5466-20 (4%)","X-8528-20 (5%)","X-8637-20 (5%)"]},"chartOpts":{"pointcolor":["#ff4136","#2ecc40","#b10dc9"],"pointsize":4,"xlab":"Average X chr intensity","ylab":"Proportion het on X chr"}},"evals":[],"jsHooks":[]}</script>
<p>In the above scatterplot, we show the proportion of hets vs the average intensity for the X chromosome SNPs. In calculating the proportion of heterozygous genotypes for the individuals, we look at X chromosome genotypes equal to 2 which corresponds to the heterozygote) relative to not being 0 (which is used to encode missing genotypes). The genotypes are arranged with rows being individuals and columns being markers.</p>
<p>The following are the mice that have had sex incorrectly assigned:</p>
<table class="table table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;">
Neogen_Sample_ID
</th>
<th style="text-align:center;">
Genoprobs_Sample_ID
</th>
<th style="text-align:center;">
Sex
</th>
<th style="text-align:center;">
Inferred.Sex
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;width: 5cm; ">
B6-1622
</td>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210604_B6-1622_B12
</td>
<td style="text-align:center;width: 5cm; ">
</td>
<td style="text-align:center;width: 5cm; ">
M
</td>
</tr>
</tbody>
</table>
</div>
<div id="sample-duplicates" class="section level3">
<h3>Sample Duplicates</h3>
<pre class="r"><code>cg &lt;- compare_geno(gm, cores=10)
summary.cg &lt;- summary(cg)</code></pre>
<table class="table table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:left;">
iid1
</th>
<th style="text-align:left;">
ind1
</th>
<th style="text-align:center;">
pc.miss.ind1
</th>
<th style="text-align:center;">
iid2
</th>
<th style="text-align:center;">
ind2
</th>
<th style="text-align:center;">
pc.miss.ind2
</th>
<th style="text-align:center;">
prop_match
</th>
<th style="text-align:center;">
n_mismatch
</th>
<th style="text-align:left;">
n_typed
</th>
<th style="text-align:left;">
n_match
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;width: 5cm; ">
B6-1622
</td>
<td style="text-align:left;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210604_B6-1622_B12
</td>
<td style="text-align:center;width: 5cm; ">
36.82772
</td>
<td style="text-align:center;width: 5cm; ">
X-8503-20
</td>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-8503-20_C12
</td>
<td style="text-align:center;width: 5cm; ">
34.09197
</td>
<td style="text-align:center;width: 5cm; ">
0.9732334
</td>
<td style="text-align:center;width: 5cm; ">
1825
</td>
<td style="text-align:left;">
68182
</td>
<td style="text-align:left;">
66357
</td>
</tr>
</tbody>
</table>
<p>Here is a histogram of the proportion of matching genotypes. The tick marks below the histogram indicate individual pairs.</p>
<pre class="r"><code>save(summary.cg,file = &quot;data/summary.cg_wcont_redone.RData&quot;)

#pdf(file = &quot;output/Proportion_matching_genotypes_before_removal_of_bad_samples_wcont_redone.pdf&quot;, width = 20, height = 20) 
#par(mar=c(5.1,0.6,0.6, 0.6))
#hist(cg[upper.tri(cg)], breaks=seq(0, 1, length=201),
#     main=&quot;&quot;, yaxt=&quot;n&quot;, ylab=&quot;&quot;, xlab=&quot;Proportion matching genotypes&quot;)
#rug(cg[upper.tri(cg)])
#dev.off()

par(mar=c(5.1,0.6,0.6, 0.6))
hist(cg[upper.tri(cg)], breaks=seq(0, 1, length=201),
     main=&quot;&quot;, yaxt=&quot;n&quot;, ylab=&quot;&quot;, xlab=&quot;Proportion matching genotypes&quot;)
rug(cg[upper.tri(cg)])</code></pre>
<p><img src="figure/2.1_sample_bqc_wcont_redone.Rmd/Sample%20duplicates%20figures-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The majority of pairs share about 50% genotypes, but there is a group of pairs that are more closely related and share about 70% genotypes. The pairs cited above have &gt;90% matching genotypes.</p>
</div>
<div id="array-intensities" class="section level3">
<h3>Array Intensities</h3>
<pre class="r"><code>#load the intensities.fst_wcont_redone.RData
#load(&quot;data/intensities.fst_wcont_redone.RData&quot;)

xn &lt;- x[,as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), &quot;_&quot;))[,6])]
xn &lt;- xn[snps$marker,]
xnm &lt;- rownames(xn)

yn &lt;- y[,as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), &quot;_&quot;))[,6])]
yn &lt;- yn[snps$marker,]

# bring together in one matrix
result &lt;- cbind(snp=rep(snps$marker, 2),
                channel=rep(c(&quot;x&quot;, &quot;y&quot;), each=length(snps$marker)),
                as.data.frame(rbind(xn, yn)))
rownames(result) &lt;- 1:nrow(result)

# bring SNP rows together
result &lt;- result[as.numeric(t(cbind(seq_along(snps$marker), seq_along(snps$marker)+length(snps$marker)))),]
rownames(result) &lt;- 1:nrow(result)

#load the intensities.fst_wcont_redone.RData
#load(&quot;data/heh/intensities.fst_wcont_redone.RData&quot;)
#X and Y channel
X &lt;- result[result$channel == &quot;x&quot;,]
rownames(X) &lt;- X$snp
X &lt;- X[,c(-1,-2)]

Y &lt;- result[result$channel == &quot;y&quot;,]
rownames(Y) &lt;- Y$snp
Y &lt;- Y[,c(-1,-2)]

int &lt;- result

#int &lt;- result

#rm(result)
int &lt;- int[seq(1, nrow(int), by=2),-(1:2)] + int[-seq(1, nrow(int), by=2),-(1:2)]
int &lt;- int[,intersect(as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), &quot;_&quot;))[,6]), colnames(int))]
names(percent_missing) &lt;- as.character(do.call(rbind.data.frame, strsplit(names(percent_missing), &quot;_&quot;))[,6])
n &lt;- names(sort(percent_missing[intersect(as.character(do.call(rbind.data.frame, strsplit(ind_ids(gm), &quot;_&quot;))[,6]), colnames(int))], decreasing=TRUE))
iboxplot(log10(t(int[,n])+1), orderByMedian=FALSE, chartOpts=list(ylab=&quot;log10(SNP intensity + 1)&quot;))</code></pre>
<p><div id="htmlwidget-0c7e572f914cbee065a8" style="width:912px;height:576px;" class="iboxplot html-widget"></div>
<script type="application/json" data-for="htmlwidget-0c7e572f914cbee065a8">{"x":{"data":{"ind":["B6-1622","X-8503-20","X-5190-20","X-5150-20","X-5443-20","X-5455-20","X-8306-20","X-8572-20","X-5232-20","X-8586-20","X-5169-20","X-5448-20","X-8294-20","X-8585-20","X-8296-20","X-5181-20","X-8614-20","X-8528-20","X-8610-20","X-2173-20","X-8637-20","X-8303-20","X-2198-20","X-5548-20","X-5187-20","X-5490-20","X-8544-20","X-5509-20","X-5499-20","X-8578-20","X-5195-20","X-2178-20","X-5466-20","X-5198-20","X-5507-20","X-5244-20","X-8560-20","X-5467-20","X-8542-20","X-8575-20","X-8577-20","X-8638-20","X-5237-20"],"qu":[0.001,0.01,0.1,0.25,0.5,0.75,0.9,0.99,0.999],"breaks":[0,0.010908,0.021816,0.032724,0.043633,0.054541,0.065449,0.076357,0.087265,0.098173,0.10908,0.11999,0.1309,0.14181,0.15271,0.16362,0.17453,0.18544,0.19635,0.20726,0.21816,0.22907,0.23998,0.25089,0.2618,0.2727,0.28361,0.29452,0.30543,0.31634,0.32724,0.33815,0.34906,0.35997,0.37088,0.38179,0.39269,0.4036,0.41451,0.42542,0.43633,0.44723,0.45814,0.46905,0.47996,0.49087,0.50178,0.51268,0.52359,0.5345,0.54541,0.55632,0.56722,0.57813,0.58904,0.59995,0.61086,0.62177,0.63267,0.64358,0.65449,0.6654,0.67631,0.68721,0.69812,0.70903,0.71994,0.73085,0.74176,0.75266,0.76357,0.77448,0.78539,0.7963,0.8072,0.81811,0.82902,0.83993,0.85084,0.86174,0.87265,0.88356,0.89447,0.90538,0.91629,0.92719,0.9381,0.94901,0.95992,0.97083,0.98173,0.99264,1.0036,1.0145,1.0254,1.0363,1.0472,1.0581,1.069,1.0799,1.0908,1.1017,1.1126,1.1235,1.1344,1.1454,1.1563,1.1672,1.1781,1.189,1.1999,1.2108,1.2217,1.2326,1.2435,1.2544,1.2653,1.2763,1.2872,1.2981,1.309,1.3199,1.3308,1.3417,1.3526,1.3635,1.3744,1.3853,1.3962,1.4072,1.4181,1.429,1.4399,1.4508,1.4617,1.4726,1.4835,1.4944,1.5053,1.5162,1.5271,1.5381,1.549,1.5599,1.5708,1.5817,1.5926,1.6035,1.6144,1.6253,1.6362,1.6471,1.658,1.6689,1.6799,1.6908,1.7017,1.7126,1.7235,1.7344,1.7453,1.7562,1.7671,1.778,1.7889,1.7998,1.8108,1.8217,1.8326,1.8435,1.8544,1.8653,1.8762,1.8871,1.898,1.9089,1.9198,1.9307,1.9417,1.9526,1.9635,1.9744,1.9853,1.9962,2.0071,2.018,2.0289,2.0398,2.0507,2.0616,2.0726,2.0835,2.0944,2.1053,2.1162,2.1271,2.138,2.1489,2.1598,2.1707,2.1816,2.1925,2.2034,2.2144,2.2253,2.2362,2.2471,2.258,2.2689,2.2798,2.2907,2.3016,2.3125,2.3234,2.3343,2.3453,2.3562,2.3671,2.378,2.3889,2.3998,2.4107,2.4216,2.4325,2.4434,2.4543,2.4652,2.4762,2.4871,2.498,2.5089,2.5198,2.5307,2.5416,2.5525,2.5634,2.5743,2.5852,2.5961,2.6071,2.618,2.6289,2.6398,2.6507,2.6616,2.6725,2.6834,2.6943,2.7052,2.7161,2.727,2.7379],"quant":[[0.0034605,0,0.029384,0.025715,0.020361,0.022841,0.0306,0.012837,0.01536,0.032216,0.026533,0.027757,0.032216,0.028571,0.028164,0.027757,0.031145,0.029384,0.028571,0.02735,0.02735,0.031004,0.027757,0.032619,0.025039,0.019532,0.022428,0.023252,0.017451,0.029789,0.019532,0.021189,0.019673,0.01494,0.017033,0.017868,0.015085,0.016616,0.01494,0.015779,0.016616,0.016616,0.012415],[0.026125,0.028164,0.044932,0.042182,0.03862,0.037825,0.049993,0.055378,0.043362,0.053463,0.047275,0.051153,0.054996,0.049606,0.053078,0.051538,0.054613,0.050766,0.052694,0.05038,0.048442,0.058805,0.052694,0.06558,0.060123,0.047275,0.058046,0.053846,0.046495,0.064083,0.047664,0.058426,0.054766,0.052309,0.053463,0.058046,0.052309,0.057286,0.057666,0.056524,0.063333,0.057666,0.059185],[0.065206,0.11093,0.081347,0.081347,0.10551,0.087071,0.11261,0.18184,0.13513,0.12743,0.11494,0.1271,0.13767,0.13066,0.1383,0.1271,0.14145,0.13066,0.14145,0.13354,0.13162,0.14737,0.13799,0.16017,0.15866,0.14145,0.15746,0.14706,0.14983,0.16791,0.15045,0.17143,0.17026,0.17056,0.16879,0.17522,0.17348,0.17725,0.18639,0.18127,0.19783,0.18184,0.19921],[0.095866,0.1685,0.13098,0.1383,0.19173,0.16406,0.18808,0.25983,0.22246,0.20683,0.19451,0.20871,0.21748,0.21192,0.21932,0.20249,0.22298,0.21192,0.2235,0.2159,0.21696,0.22505,0.21932,0.22866,0.22891,0.22634,0.22943,0.22608,0.23325,0.24304,0.23198,0.24724,0.24674,0.24601,0.24773,0.25237,0.2492,0.24748,0.25864,0.25575,0.26435,0.256,0.26623],[0.14489,0.25672,0.24403,0.26293,0.31806,0.29314,0.29951,0.33486,0.32838,0.31112,0.30707,0.31366,0.3145,0.31345,0.3166,0.31112,0.3166,0.31302,0.31765,0.31806,0.31534,0.31827,0.31827,0.31597,0.31744,0.32284,0.32035,0.32098,0.32449,0.32222,0.32035,0.3249,0.32469,0.32531,0.32654,0.32858,0.32346,0.32222,0.32654,0.32756,0.32675,0.32695,0.32736],[0.23325,0.38399,0.43249,0.4442,0.44404,0.43265,0.41514,0.40106,0.42716,0.40943,0.41514,0.41128,0.40432,0.40773,0.40722,0.4239,0.40006,0.40671,0.40157,0.41246,0.40226,0.40364,0.4079,0.4128,0.41731,0.40875,0.41697,0.41128,0.40518,0.39322,0.39985,0.3948,0.3941,0.39967,0.39637,0.39528,0.39199,0.39375,0.38757,0.39146,0.38274,0.39005,0.38328],[0.4138,0.55642,0.63709,0.61162,0.5463,0.54518,0.51508,0.4603,0.51041,0.49052,0.50732,0.49457,0.47958,0.48869,0.48558,0.51878,0.47026,0.48686,0.4729,0.49206,0.47741,0.47712,0.4843,0.5061,0.5193,0.48219,0.50637,0.49108,0.47625,0.45697,0.47144,0.45834,0.45484,0.47041,0.45909,0.45576,0.4572,0.4624,0.44483,0.45133,0.43584,0.44917,0.43696],[1.2476,1.2732,0.94817,0.8522,0.69936,0.70919,0.6602,0.56761,0.64296,0.61395,0.63619,0.61584,0.60314,0.62052,0.62325,0.66941,0.57496,0.61363,0.58047,0.61643,0.58984,0.59517,0.59999,0.69046,0.71198,0.60217,0.66229,0.63195,0.59912,0.56538,0.59671,0.57403,0.55654,0.60746,0.56277,0.56526,0.58035,0.59517,0.54642,0.55558,0.53453,0.5563,0.53349],[1.8857,2.0083,1.0139,0.91806,0.75689,0.76569,0.71422,0.62936,0.70116,0.66354,0.68768,0.6682,0.65437,0.67394,0.67645,0.72263,0.61937,0.66121,0.6291,0.67207,0.63686,0.64424,0.65228,0.74948,0.78608,0.65115,0.71464,0.68895,0.68018,0.61742,0.67596,0.63029,0.60595,0.67815,0.61947,0.62249,0.64962,0.65312,0.60387,0.6066,0.58354,0.61045,0.58497]],"counts":[[343,643,1331,2430,3794,5322,6693,7413,8160,8071,8090,7474,7196,6293,5680,5134,4405,4038,3540,3164,2760,2499,2217,1967,1749,1598,1428,1309,1200,1138,1037,956,875,836,792,746,694,643,587,596,529,548,494,476,435,425,395,374,333,357,327,263,306,243,232,233,238,209,225,183,167,154,182,161,148,150,146,120,112,112,105,112,104,94,117,94,87,88,71,91,87,78,69,60,80,84,54,63,77,67,54,50,54,60,55,59,49,47,43,55,48,50,51,61,52,55,42,42,40,36,54,43,36,32,34,34,30,31,39,29,36,29,33,38,22,29,26,35,33,25,21,19,34,29,25,25,14,31,20,20,21,14,22,17,18,24,18,19,18,15,26,20,23,16,21,10,22,11,17,21,14,16,7,10,15,14,16,9,11,7,7,3,6,4,4,9,6,7,4,3,3,7,4,5,5,7,1,3,0,1,2,0,3,3,3,4,3,1,6,3,1,2,2,1,1,1,2,0,1,0,0,1,0,2,2,1,1,4,1,1,1,0,1,2,0,1,1,3,0,0,1,0,0,0,0,0,0,0,1,1,0,0,0,0,0,0,0,0,0,0,0],[608,469,555,671,980,1201,1612,1951,2390,2849,3210,3532,4022,4052,4237,4410,4535,4314,4520,4372,4161,4103,3931,3808,3684,3575,3290,3113,3093,2805,2776,2549,2509,2396,2278,2126,2017,1895,1667,1581,1508,1468,1365,1189,1125,1048,969,852,815,724,702,669,622,601,564,488,465,471,419,398,384,347,345,370,309,325,301,285,265,214,221,232,235,210,185,176,175,182,162,152,123,132,130,134,131,98,111,93,94,85,95,84,77,68,77,70,85,63,57,70,41,60,51,50,38,46,46,40,53,36,31,40,36,36,44,40,35,33,24,39,34,29,28,22,32,30,23,29,35,28,31,30,25,26,26,18,25,17,20,23,21,20,17,19,23,12,17,12,15,14,19,16,16,11,18,16,19,9,16,15,20,15,6,20,16,10,15,13,8,13,9,10,13,10,5,9,7,14,14,11,8,12,13,5,8,10,5,3,6,4,5,9,5,2,6,3,5,3,4,3,3,2,5,0,1,3,7,2,3,4,0,1,1,1,2,3,3,1,0,0,2,0,3,0,0,0,2,1,0,1,1,1,0,0,0,0,1,1,1,0,0,0,1,0,0,0,0,1,0,0,1],[6,25,224,947,2212,3654,4501,4677,4665,4736,4486,4172,4142,3882,3680,3457,3337,3301,3124,3037,2796,2667,2550,2589,2379,2411,2314,2186,2129,2054,1981,1909,1818,1844,1704,1692,1676,1579,1537,1393,1439,1410,1325,1313,1301,1219,1192,1199,1179,1052,1036,978,1030,891,877,831,812,789,735,699,645,637,682,615,647,600,531,482,472,433,423,398,394,406,361,361,321,318,315,315,300,276,263,264,235,264,258,260,229,216,220,171,129,72,40,12,0,1,4,1,1,0,0,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[8,57,336,1182,2446,3566,4166,4190,4057,4019,4031,3770,3783,3495,3484,3242,3120,3049,3021,2775,2778,2693,2592,2552,2505,2488,2371,2286,2275,2169,2141,2101,2054,1932,1990,1882,1902,1802,1754,1763,1648,1638,1579,1611,1523,1496,1423,1431,1325,1257,1371,1214,1132,1068,1006,989,973,840,894,817,829,782,704,628,590,568,552,537,478,479,412,400,363,340,301,316,314,293,261,264,230,223,156,124,65,48,21,10,1,2,1,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[24,161,607,1236,1697,1877,2099,2212,2334,2336,2492,2474,2621,2556,2701,2696,2624,2728,2915,2959,2843,2923,2858,2967,3005,3030,3028,3079,3103,3095,3120,3065,3057,3030,2967,3007,2931,2907,2850,2860,2714,2626,2527,2386,2363,2240,2108,1961,1841,1630,1583,1438,1369,1167,1068,1014,900,806,686,606,588,483,428,332,323,306,244,200,143,110,45,28,7,3,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[14,93,598,1696,2648,2836,3024,2889,2944,2932,2984,2926,2935,2874,2877,2882,2884,2900,2993,2961,2837,2939,2823,2841,2808,2986,2919,2871,2744,2766,2833,2711,2801,2735,2733,2527,2576,2575,2479,2415,2443,2402,2176,2217,2021,1986,1763,1670,1609,1559,1435,1325,1211,1112,1001,904,856,772,686,640,598,508,508,426,336,333,282,251,202,149,92,43,19,7,3,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[4,33,160,524,1172,1710,2060,2171,2437,2598,2678,2761,2928,2948,3051,3137,3171,3232,3190,3318,3302,3360,3363,3442,3438,3449,3474,3425,3373,3355,3466,3247,3299,3291,3048,3143,3085,3009,2792,2656,2553,2435,2292,2152,2014,1828,1671,1577,1471,1366,1247,1149,1033,885,880,726,662,556,507,484,355,307,283,242,172,118,53,20,6,3,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[100,207,288,334,402,509,555,650,697,813,947,1115,1234,1392,1498,1704,1903,2122,2393,2641,2916,3191,3556,3862,4102,4518,4852,5096,5447,5562,5822,5779,5907,5899,5822,5334,4915,4741,4166,3795,3303,2927,2485,2291,1847,1536,1251,1026,837,667,492,436,346,274,236,161,144,110,46,34,23,8,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[59,242,437,649,910,1083,1302,1380,1534,1617,1883,1855,2060,2162,2374,2537,2576,2751,2799,2936,3118,3096,3239,3415,3455,3673,3664,3795,3740,3885,3965,3984,3972,3893,3862,3773,3674,3558,3510,3460,3174,2984,2873,2550,2373,2239,1908,1784,1675,1498,1236,1079,1010,887,779,640,558,477,382,330,282,279,184,109,87,45,19,12,5,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[6,27,120,393,940,1268,1636,1700,1893,2041,2226,2281,2534,2522,2749,2864,2971,3037,3261,3210,3300,3466,3446,3617,3717,3651,3929,3918,3804,3983,3827,3906,3880,3847,3862,3736,3637,3382,3339,3200,2879,2709,2522,2288,2144,1927,1719,1517,1330,1211,1020,926,760,720,571,487,399,344,313,189,139,67,26,11,7,0,1,0,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[12,64,241,700,1231,1622,1927,1986,2196,2421,2549,2585,2780,2802,2777,3004,3003,2951,3017,3099,3153,3321,3293,3361,3458,3451,3570,3600,3444,3594,3584,3547,3602,3575,3401,3387,3316,3168,3097,2844,2706,2527,2447,2260,2132,1915,1789,1667,1500,1350,1254,1208,965,881,783,664,596,480,385,326,281,199,153,89,40,15,5,0,0,1,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[12,40,199,520,943,1219,1534,1692,1871,1995,2244,2253,2477,2523,2761,2791,2762,3042,3044,3101,3270,3399,3431,3628,3579,3823,3721,3813,3872,3960,3878,3890,4039,3862,3953,3636,3737,3544,3279,3147,2869,2756,2504,2231,2107,1925,1740,1614,1449,1276,1176,978,812,755,591,504,429,330,284,221,142,100,46,18,6,1,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[6,17,135,405,765,999,1242,1391,1566,1807,1910,1984,2374,2359,2606,2669,2893,3029,3106,3245,3409,3537,3675,3825,3832,4029,4015,4256,4318,4366,4280,4203,4317,4188,4234,3912,3817,3765,3482,3237,2953,2777,2492,2259,1922,1806,1582,1353,1186,1000,914,752,632,553,473,393,343,288,205,135,74,35,12,7,4,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[4,36,243,594,945,1185,1422,1589,1733,1861,2008,2140,2447,2460,2497,2755,2940,3066,3104,3277,3248,3514,3491,3729,3673,3819,3949,3948,4024,4202,4047,4076,4075,3967,4011,3769,3781,3596,3269,3129,2895,2632,2444,2346,2044,1809,1658,1452,1276,1133,1004,904,774,606,561,460,384,333,313,231,209,128,64,26,18,4,0,0,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[5,41,206,510,731,1017,1224,1371,1562,1712,1886,2015,2188,2296,2507,2636,2810,3045,3106,3119,3387,3518,3655,3819,3844,4031,4018,4116,4214,4322,4270,4332,4203,4142,4238,3955,3743,3640,3482,3157,3030,2693,2480,2224,2084,1802,1612,1388,1239,1046,884,790,697,584,504,424,378,332,304,276,217,145,74,37,17,7,4,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[13,49,219,509,802,1175,1469,1584,1930,2152,2334,2410,2739,2766,2965,2993,3177,3179,3261,3332,3358,3490,3504,3487,3467,3514,3562,3451,3490,3467,3582,3425,3466,3315,3250,3276,3195,3068,2976,2908,2770,2598,2468,2412,2230,2039,1842,1683,1512,1376,1299,1147,1062,920,854,719,651,574,542,459,378,363,306,304,204,160,84,46,27,5,3,0,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[10,33,135,463,726,964,1159,1309,1486,1604,1839,1943,2177,2283,2359,2588,2809,2865,3008,3151,3348,3551,3667,3909,3990,4173,4231,4327,4445,4488,4661,4586,4557,4561,4521,4240,4131,3891,3643,3344,3050,2680,2449,2186,1951,1808,1440,1327,1173,975,771,624,515,391,354,225,136,74,28,10,1,1,2,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[6,28,207,547,919,1161,1429,1529,1746,1908,2102,2167,2396,2483,2567,2760,2847,3003,3164,3254,3404,3517,3576,3741,3691,3800,3866,4070,4083,4087,4157,4043,4160,4095,3935,3762,3735,3631,3298,3193,2894,2606,2470,2235,2095,1858,1618,1480,1228,1132,966,855,753,645,534,433,399,366,259,214,138,53,29,12,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[8,45,225,465,759,1048,1163,1263,1461,1608,1751,1899,2131,2232,2414,2520,2729,2879,3040,3077,3271,3549,3657,3779,3991,4187,4249,4235,4534,4577,4625,4534,4565,4502,4428,4206,4183,3884,3583,3380,2927,2664,2510,2320,1981,1803,1546,1363,1170,960,796,675,517,407,356,300,177,120,65,33,10,4,1,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[8,54,253,532,893,1171,1281,1501,1596,1770,2005,2106,2294,2357,2610,2675,2840,2903,3088,3064,3251,3361,3449,3625,3658,3806,3963,3876,4044,4057,4100,4026,4079,4036,3928,3942,3748,3764,3514,3243,2944,2782,2593,2431,2168,1845,1684,1546,1438,1253,1128,957,753,670,580,497,418,310,271,222,170,111,56,28,11,4,2,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[11,47,272,635,963,1127,1392,1551,1720,1848,1953,2073,2281,2324,2498,2547,2702,2741,2900,3140,3250,3339,3558,3688,3782,4074,4078,4216,4372,4264,4514,4468,4543,4330,4236,4128,3865,3824,3456,3216,2860,2632,2428,2263,1977,1792,1577,1370,1244,1086,882,750,654,481,420,305,276,214,112,57,15,6,2,0,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[8,36,130,364,535,806,982,1174,1411,1570,1652,1860,2130,2248,2456,2590,2737,3000,3178,3308,3440,3533,3639,3832,4072,4084,4278,4344,4476,4660,4616,4462,4592,4430,4232,4108,4029,3834,3596,3322,3066,2701,2445,2150,1982,1784,1550,1346,1192,978,880,747,667,490,449,342,321,208,156,83,41,14,5,1,0,0,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[17,36,243,483,727,1003,1216,1341,1584,1729,1879,2039,2234,2303,2541,2681,2792,2998,3049,3156,3345,3329,3393,3705,3818,3909,4124,4155,4071,4278,4291,4233,4300,4214,4175,4076,3961,3761,3559,3285,3055,2745,2520,2356,2089,1892,1738,1551,1292,1118,1013,818,731,548,469,409,276,253,193,131,61,34,14,5,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[3,22,119,266,424,518,626,801,970,1175,1501,1626,1967,2143,2388,2670,3008,3182,3439,3797,3795,3910,4237,4319,4186,4477,4428,4381,4386,4384,4203,4056,4125,3872,3857,3681,3344,3261,3138,2888,2697,2515,2300,2072,1969,1812,1639,1415,1250,1129,1017,900,794,761,627,561,498,506,433,390,339,344,335,314,290,242,219,165,127,54,37,12,2,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[20,79,221,337,444,557,661,889,977,1240,1435,1578,1931,2118,2376,2678,2864,3070,3478,3577,3844,3920,4122,4232,4172,4419,4447,4217,4354,4279,4176,4042,3970,3854,3691,3536,3357,3161,2899,2852,2588,2464,2255,2093,1882,1761,1538,1525,1368,1218,1155,963,863,775,741,632,629,556,482,490,442,397,343,348,316,280,241,192,206,159,145,85,70,39,22,8,2,1,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[19,167,377,581,792,1012,1142,1302,1361,1567,1743,1758,1993,2123,2278,2438,2534,2769,2910,3087,3187,3289,3476,3735,3735,4032,4146,4193,4331,4425,4425,4479,4421,4328,4420,4309,4173,3977,3678,3514,3201,2825,2565,2356,2053,1858,1654,1447,1239,1062,935,812,654,551,445,412,353,276,178,139,56,35,14,2,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[19,110,274,344,458,542,690,841,1033,1221,1452,1708,1901,2175,2399,2703,2821,3059,3302,3536,3622,3806,3882,4210,4143,4126,4265,4182,4276,4312,4187,4113,4062,3977,3891,3600,3614,3462,3195,3016,2836,2658,2461,2313,2102,1928,1734,1542,1377,1262,1024,954,868,819,703,651,567,472,511,399,367,357,289,248,173,106,64,17,11,2,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[11,101,287,385,623,784,985,1102,1293,1532,1682,1845,2106,2138,2416,2532,2675,2986,3111,3217,3459,3474,3757,3875,3886,4038,4114,4170,4248,4361,4351,4441,4293,4231,4243,3871,3846,3601,3480,3270,2961,2829,2557,2300,2101,1835,1584,1355,1335,1122,985,888,843,628,564,523,427,338,354,296,234,178,137,88,36,20,12,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[39,218,374,557,709,879,975,1081,1189,1410,1491,1615,1787,1937,2174,2289,2475,2693,2880,2998,3186,3493,3627,3953,3881,4118,4284,4432,4490,4607,4861,4818,4783,4675,4701,4490,4252,4000,3713,3468,3088,2806,2637,2223,1984,1666,1552,1355,1155,926,864,713,589,437,399,305,219,186,172,129,101,79,57,39,30,11,10,4,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[6,34,153,296,416,539,689,790,910,1071,1285,1376,1637,1770,1968,2142,2383,2682,2870,3062,3432,3749,3853,4242,4390,4754,4872,5169,5214,5440,5497,5398,5586,5213,5112,4749,4518,4032,3737,3286,2914,2636,2319,1971,1721,1441,1209,950,798,737,583,436,359,318,258,163,108,60,25,14,5,0,1,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[30,172,366,535,701,825,928,1074,1221,1375,1521,1677,1850,1927,2200,2341,2569,2753,2888,3141,3271,3608,3825,3979,3946,4352,4539,4582,4711,4889,4752,4973,4879,4747,4709,4346,4177,3914,3635,3232,3033,2595,2330,2149,1879,1602,1431,1205,1054,858,782,659,524,443,369,282,216,197,153,122,106,69,60,32,21,13,4,4,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[30,120,267,363,433,485,635,729,826,1014,1159,1309,1454,1641,1801,2049,2271,2547,2807,3008,3298,3494,3904,4194,4260,4737,4968,5141,5353,5485,5692,5574,5649,5358,5226,4840,4494,4309,3669,3373,2980,2633,2338,1963,1732,1451,1233,974,836,662,549,432,366,322,304,217,159,120,53,35,16,7,1,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[32,163,332,399,436,556,604,727,812,1006,1173,1234,1511,1677,1858,2036,2290,2482,2700,3107,3286,3518,3882,4137,4327,4776,4940,5185,5177,5567,5511,5568,5675,5394,5292,4945,4740,4320,3874,3398,3035,2727,2308,2020,1690,1490,1153,920,794,663,513,419,337,246,176,106,51,27,16,2,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[64,208,322,394,481,558,588,687,815,963,1210,1254,1470,1613,1820,2053,2303,2580,2812,2998,3337,3612,3896,4118,4363,4717,4878,5014,5194,5251,5400,5280,5261,5020,4937,4711,4322,4117,3598,3270,2954,2686,2234,1994,1741,1545,1343,1184,1016,836,741,592,537,496,373,311,279,227,195,173,128,114,65,39,30,14,4,1,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[49,219,318,338,487,534,710,784,899,1018,1232,1218,1493,1653,1869,1945,2249,2415,2599,2917,3231,3483,3794,4096,4247,4561,4840,5102,5275,5414,5548,5511,5594,5570,5280,4941,4614,4272,3818,3418,3125,2694,2307,2157,1794,1591,1282,1033,873,706,574,436,349,280,200,152,119,62,30,19,8,0,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[45,152,236,348,442,500,642,720,760,915,1099,1142,1343,1545,1779,1913,2106,2343,2545,2944,3126,3440,3685,3982,4312,4729,4973,5178,5481,5583,5806,5728,5792,5697,5640,5132,4828,4386,3929,3480,3087,2687,2328,2024,1768,1396,1083,964,766,599,531,396,335,286,260,158,108,78,40,17,10,3,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[68,225,341,386,459,523,586,591,766,1001,1111,1236,1398,1514,1728,1999,2228,2460,2625,2965,3286,3566,3883,4259,4472,4930,5242,5470,5595,5680,5906,5731,5533,5465,5209,4836,4379,4049,3616,3167,2880,2429,2178,1855,1632,1423,1132,1007,747,671,592,480,400,309,271,217,182,147,119,80,55,29,18,6,2,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[53,185,292,318,413,448,498,576,744,810,1045,1156,1323,1521,1775,2029,2351,2587,2916,3211,3499,3778,4016,4523,4681,4999,5208,5324,5414,5668,5594,5523,5403,5285,4828,4564,4202,3974,3551,3187,2852,2452,2058,1880,1638,1387,1203,1076,872,807,609,544,506,393,385,312,274,210,166,126,64,35,14,9,2,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[67,221,265,315,387,414,457,558,622,724,887,988,1205,1255,1425,1731,2011,2149,2463,2749,3142,3510,3864,4273,4672,5087,5433,5753,6183,6325,6489,6445,6375,6186,5530,5204,4749,4257,3690,3271,2880,2294,2123,1797,1484,1200,983,774,640,422,385,297,247,163,138,85,44,31,17,2,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[60,198,300,347,391,446,517,581,646,873,935,1103,1229,1409,1623,1826,2088,2210,2545,2860,3030,3478,3748,4273,4498,4842,5198,5508,5752,6034,6191,6132,6123,5788,5581,5218,4680,4268,3814,3356,2965,2509,2242,1908,1615,1344,1054,873,744,556,483,366,331,257,171,130,47,28,15,5,0,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[48,192,267,295,301,339,398,430,503,580,674,760,1042,1165,1297,1491,1724,1942,2241,2674,3103,3452,3836,4449,4739,5403,5893,6220,6609,6943,7140,6994,6850,6468,6031,5397,4867,4263,3596,3127,2653,2233,1798,1501,1199,982,729,613,515,392,333,261,182,119,47,29,10,2,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[60,170,289,348,361,476,565,565,680,842,938,1092,1285,1400,1573,1783,1958,2234,2479,2822,3160,3420,3893,4269,4539,4875,5220,5687,5749,6105,6149,6060,6340,5884,5688,5200,4746,4241,3830,3296,2891,2455,2144,1790,1597,1274,1029,856,676,563,435,347,323,252,185,129,65,37,15,14,3,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],[115,218,264,301,328,342,380,427,460,565,686,795,944,1079,1272,1389,1709,1917,2234,2536,2956,3319,3754,4378,4809,5247,5950,6225,6787,7239,7214,6985,6857,6466,6026,5377,4840,4190,3712,3135,2662,2304,1950,1576,1204,995,783,612,465,379,328,246,172,114,53,29,16,4,0,1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]]},"chartOpts":{"ylab":"log10(SNP intensity + 1)"}},"evals":[],"jsHooks":[]}</script> In the above plot, distributions of array intensities (after a log10(x+1) transformation) are displayed.</p>
<p>The arrays are sorted by the proportion of missing genotype data for the sample, and the curves connect various quantiles of the intensities.</p>
<pre class="r"><code>qu &lt;- apply(int, 2, quantile, c(0.01, 0.99), na.rm=TRUE)
group &lt;- (percent_missing &gt;= 19.97) + (percent_missing &gt; 5) + (percent_missing &gt; 2) + 1
labels &lt;- paste0(colnames(qu), &quot; (&quot;, round(percent_missing), &quot;%)&quot;)
iplot(qu[1,], qu[2,], indID=labels, group=group,
      chartOpts=list(xlab=&quot;1 %ile of array intensities&quot;,
                     ylab=&quot;99 %ile of array intensities&quot;,
                     pointcolor=c(&quot;#ccc&quot;, &quot;slateblue&quot;, &quot;Orchid&quot;, &quot;#ff851b&quot;)))</code></pre>
<p><div id="htmlwidget-859faef2725068e45d18" style="width:912px;height:576px;" class="iplot html-widget"></div>
<script type="application/json" data-for="htmlwidget-859faef2725068e45d18">{"x":{"data":{"x":[0.14848,0.141,0.142,0.142,0.109,0.115,0.143,0.116,0.113,0.128,0.128,0.131,0.136,0.105,0.132,0.139,0.146,0.163,0.157,0.123,0.143,0.135,0.159,0.144,0.093,0.13,0.121,0.129,0.125,0.145,0.131,0.102,0.091,0.122,0.129,0.115,0.062,0.067,0.134,0.126,0.1344,0.124,0.118],"y":[4.152,2.937,2.519,2.6,7.875,3.001,3.595,2.951,2.973,2.805,3.05,2.654,2.695,3.395,3.285,2.594,2.4158,3.903,2.424,3.1346,2.675,3.01,2.676,2.75,4.0045,3.2,3.1737,2.981,3.129,2.937,3.111,6.1155,4.119,3.573,2.806,3.327,16.683,17.759,2.758,3.671,2.602,3.108,2.889],"group":[2,2,2,2,4,2,2,2,2,2,2,2,3,3,2,2,2,2,2,2,2,3,2,2,3,3,3,2,3,2,3,4,3,3,3,3,4,4,3,3,2,3,2],"indID":["X-5187-20 (4%)","X-5467-20 (4%)","X-8542-20 (4%)","X-8638-20 (3%)","X-5190-20 (33%)","X-5490-20 (4%)","X-8544-20 (4%)","X-5195-20 (4%)","X-5499-20 (4%)","X-8560-20 (4%)","X-5198-20 (4%)","X-5507-20 (4%)","X-8572-20 (8%)","X-5232-20 (8%)","X-5509-20 (4%)","X-8575-20 (4%)","X-5237-20 (3%)","X-5548-20 (5%)","X-8577-20 (4%)","X-2173-20 (5%)","X-5244-20 (4%)","X-8294-20 (5%)","X-8578-20 (4%)","X-2178-20 (4%)","X-5443-20 (15%)","X-8296-20 (5%)","X-8585-20 (5%)","X-2198-20 (5%)","X-5448-20 (6%)","X-8303-20 (5%)","X-8586-20 (7%)","X-5150-20 (24%)","X-5455-20 (11%)","X-8306-20 (9%)","X-8610-20 (5%)","X-5169-20 (6%)","B6-1622 (37%)","X-8503-20 (34%)","X-8614-20 (5%)","X-5181-20 (5%)","X-5466-20 (4%)","X-8528-20 (5%)","X-8637-20 (5%)"]},"chartOpts":{"xlab":"1 %ile of array intensities","ylab":"99 %ile of array intensities","pointcolor":["#ccc","slateblue","Orchid","#ff851b"]}},"evals":[],"jsHooks":[]}</script> For this particular set of arrays, a plot of the 1 %ile vs the 99 %ile is quite revealing. In the following, the orange points are those with &gt; 20% missing genotypes, the pink points are the samples with 5-20% missing genotypes, and the blue points are the samples with 2-5% missing genotypes.</p>
</div>
<div id="genotype-frequencies" class="section level3">
<h3>Genotype Frequencies</h3>
<p>The following triangle plots show the genotype frequency distributions for the mice, among the four groups of markers with common minor allele frequency (MAF) in the founder strains. These plots make use of the fact that for a point within an equilateral triangle, the sum of the distances to the three sides is a constant.</p>
<pre class="r"><code>g &lt;- do.call(&quot;cbind&quot;, gm$geno[1:19])
fg &lt;- do.call(&quot;cbind&quot;, gm$founder_geno[1:19])
g &lt;- g[,colSums(fg==0)==0]
fg &lt;- fg[,colSums(fg==0)==0]
fgn &lt;- colSums(fg==3)

gf_ind &lt;- vector(&quot;list&quot;, 4)
for(i in 1:4) {
  gf_ind[[i]] &lt;- t(apply(g[,fgn==i], 1, function(a) table(factor(a, 1:3))/sum(a != 0)))
}

par(mfrow=c(4,1), mar=c(0.6, 0.6, 2.6, 0.6))
for(i in 1:4) {
  triplot(c(&quot;AA&quot;, &quot;AB&quot;, &quot;BB&quot;), main=paste0(&quot;MAF = &quot;, i, &quot;/8&quot;))
  tripoints(gf_ind[[i]], pch=21, bg=&quot;lightblue&quot;)
  tripoints(c((1-i/8)^2, 2*i/8*(1-i/8), (i/8)^2), pch=21, bg=&quot;violetred&quot;)
  
  if(i&gt;=3) { # label mouse with lowest het
    wh &lt;- which(gf_ind[[i]][,2] == min(gf_ind[[i]][,2]))
    tritext(gf_ind[[i]][wh,,drop=FALSE] + c(0.02, -0.02, 0),
            names(wh), adj=c(0, 1))
  }
  
  # label other mice
  if(i==1) {
    lab &lt;- rownames(gf_ind[[i]])[gf_ind[[i]][,2]&gt;0.3]
  }
  else if(i==2) {
    lab &lt;- rownames(gf_ind[[i]])[gf_ind[[i]][,2]&gt;0.48]
  }
  else if(i==3) {
    lab &lt;- rownames(gf_ind[[i]])[gf_ind[[i]][,2]&gt;0.51]
  }
  else if(i==4) {
    lab &lt;- rownames(gf_ind[[i]])[gf_ind[[i]][,2]&gt;0.6]
  }
  
  for(ind in lab) {
    if(grepl(&quot;^F&quot;, ind) &amp;&amp; i != 3) {
      tritext(gf_ind[[i]][ind,,drop=FALSE] + c(-0.01, 0, +0.01), ind, adj=c(1,0.5))
    } else {
      tritext(gf_ind[[i]][ind,,drop=FALSE] + c(0.01, 0, -0.01), ind, adj=c(0,0.5))
    }
  }
}</code></pre>
<p><img src="figure/2.1_sample_bqc_wcont_redone.Rmd/Genotype%20frequencies-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>The majority of individuals are tightly clustered around the expected distribution (in pink). There are a few outliers with elevated &amp; reduced heterozygosity, but most of these are among the samples with highest rates of missing genotypes.</p>
</div>
<div id="crossover-counts" class="section level3">
<h3>Crossover Counts</h3>
<pre class="r"><code>#load pre-caluated results
#load(&quot;data/pr_wcont_redone.RData&quot;)
pr&lt;-readRDS(paste0(filepaths,&quot;43_DO_Korstanje_Redo__GigaMUGA_genoprobs_36state.rds&quot;))
#load(&quot;data/m_wcont_redone.RData&quot;)
#load(&quot;data/nxo_wcont_redone.RData&quot;)

m &lt;- maxmarg(pr)
nxo &lt;- count_xo(m)
#save(pr, file = &quot;data/pr_wcont_redone.RData&quot;)
#str(pr)
save(m, file = &quot;data/m_wcont_redone.RData&quot;)
save(nxo, file = &quot;data/nxo_wcont_redone.RData&quot;)

e &lt;- calc_errorlod(gm, pr)
#str(e)
e &lt;- do.call(&quot;cbind&quot;, e)
save(e, file = &quot;data/e_wcont_redone.RData&quot;)

snpg &lt;- predict_snpgeno(gm, m)
snpg &lt;- do.call(&quot;cbind&quot;, snpg)
save(snpg, file = &quot;data/snpg_wcont_redone.RData&quot;)

#crossover
totxo &lt;- rowSums(nxo)[rownames(gm$covar)]
names(totxo) &lt;- as.character(do.call(rbind.data.frame, strsplit(names(totxo), &quot;_&quot;))[,6])
iplot(seq_along(totxo),
      totxo,
      group=gm$covar$generation,
      chartOpts=list(xlab=&quot;Mouse&quot;, ylab=&quot;Number of crossovers&quot;, 
                     margin=list(left=80,top=40,right=40,bottom=40,inner=5),
                     axispos=list(xtitle=25,ytitle=50,xlabel=5,ylabel=5)))</code></pre>
<div id="htmlwidget-a6401eaf80dfa0282aae" style="width:912px;height:576px;" class="iplot html-widget"></div>
<script type="application/json" data-for="htmlwidget-a6401eaf80dfa0282aae">{"x":{"data":{"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43],"y":[622,662,659,680,1632,657,670,618,626,653,621,624,634,642,599,637,646,636,604,622,671,690,622,614,861,640,634,630,634,669,700,1113,730,703,656,665,4391,5946,654,620,666,625,692],"group":[3,3,4,5,3,3,4,3,3,4,3,3,4,3,3,4,3,3,4,2,3,4,4,2,3,4,4,2,3,4,4,3,3,4,5,3,1,4,5,3,3,4,5],"indID":["X-5187-20","X-5467-20","X-8542-20","X-8638-20","X-5190-20","X-5490-20","X-8544-20","X-5195-20","X-5499-20","X-8560-20","X-5198-20","X-5507-20","X-8572-20","X-5232-20","X-5509-20","X-8575-20","X-5237-20","X-5548-20","X-8577-20","X-2173-20","X-5244-20","X-8294-20","X-8578-20","X-2178-20","X-5443-20","X-8296-20","X-8585-20","X-2198-20","X-5448-20","X-8303-20","X-8586-20","X-5150-20","X-5455-20","X-8306-20","X-8610-20","X-5169-20","B6-1622","X-8503-20","X-8614-20","X-5181-20","X-5466-20","X-8528-20","X-8637-20"]},"chartOpts":{"xlab":"Mouse","ylab":"Number of crossovers","margin":{"left":80,"top":40,"right":40,"bottom":40,"inner":5},"axispos":{"xtitle":25,"ytitle":50,"xlabel":5,"ylabel":5}}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>#save crossover into pdf
pdf(file = &quot;output/number_crossover_wcont_redone.pdf&quot;)
cross_over &lt;- data.frame(Mouse = seq_along(totxo), id=names(totxo), Number_crossovers = totxo, generation = gm$covar$generation)
#names(totxo) &lt;- as.character(do.call(rbind.data.frame, strsplit(names(totxo), &quot;V01_&quot;))[,2])
#names(totxo) &lt;- as.character(do.call(rbind.data.frame, strsplit(names(totxo), &quot;_&quot;))[,6])
names(totxo)[totxo &gt;= 200 &amp; totxo &lt;= 800] = &quot;&quot;
# Change point shapes and colors
p &lt;-ggplot(cross_over, aes(x=Mouse, y=Number_crossovers, fill = generation, color=generation)) +
  geom_point() +
  geom_text_repel(aes(label=names(totxo),hjust=0,vjust=0), show.legend = FALSE)
p
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>p</code></pre>
<p><img src="figure/2.1_sample_bqc_wcont_redone.Rmd/Crossover%20counts-2.png" width="912" style="display: block; margin: auto;" /> Counts of inferred crossovers can be a useful diagnostic for identifying problem samples, which may show an excessive number of apparent crossovers. Above is a plot of the number of crossovers vs the mouse index, colored by generation.</p>
<p>We’re looking for mice with an excessive number of crossovers as well as mice with very few crossovers. These are shown below</p>
<p><strong>NB:</strong> DO generation of an individual is taken into account.</p>
<pre><code>          percent_missing total_xo
X-8610-20            5.05      656
X-8614-20            5.18      654
X-8528-20            5.18      625
X-5181-20            5.24      620
X-8296-20            5.33      640
X-8585-20            5.44      634
X-8294-20            5.49      690
X-5448-20            6.03      634
X-5169-20            6.21      665
X-8586-20            6.73      700
X-5232-20            7.51      642
X-8572-20            7.92      634
X-8306-20            8.73      703
X-5455-20           10.73      730
X-5443-20           14.98      861
X-5150-20           24.46     1113
X-5190-20           32.80     1632
X-8503-20           34.09     5946
B6-1622             36.83     4391</code></pre>
<table class="table table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;">
Neogen_Sample_ID
</th>
<th style="text-align:center;">
Genoprobs_Sample_ID
</th>
<th style="text-align:center;">
number.xo
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
X-5190-20
</td>
<td style="text-align:center;">
Jackson_Lab_Korstanje_MURGIGV01_20210701_X-5190-20_H5
</td>
<td style="text-align:center;">
1632
</td>
</tr>
<tr>
<td style="text-align:center;">
X-5150-20
</td>
<td style="text-align:center;">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-5150-20_E11
</td>
<td style="text-align:center;">
1113
</td>
</tr>
<tr>
<td style="text-align:center;">
B6-1622
</td>
<td style="text-align:center;">
Jackson_Lab_Korstanje_MURGIGV01_20210604_B6-1622_B12
</td>
<td style="text-align:center;">
4391
</td>
</tr>
<tr>
<td style="text-align:center;">
X-8503-20
</td>
<td style="text-align:center;">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-8503-20_C12
</td>
<td style="text-align:center;">
5946
</td>
</tr>
</tbody>
</table>
</div>
<div id="genotyping-error-lod-scores" class="section level3">
<h3>Genotyping Error LOD Scores</h3>
<pre class="r"><code>#load(&quot;data/e_wcont_redone.RData&quot;)
errors_ind &lt;- rowSums(e&gt;2)[rownames(gm$covar)]/n_typed(gm)*100
lab &lt;- paste0(as.character(do.call(rbind.data.frame, strsplit(names(errors_ind), &quot;_&quot;))[,6]), &quot; (&quot;, myround(percent_missing[as.character(do.call(rbind.data.frame, strsplit(rownames(gm$covar), &quot;_&quot;))[,6])],1), &quot;%)&quot;)
iplot(seq_along(errors_ind), errors_ind, indID=lab,
      chartOpts=list(xlab=&quot;Mouse&quot;, ylab=&quot;Percent genotyping errors&quot;, ylim=c(0, 15),
                     axispos=list(xtitle=25, ytitle=50, xlabel=5, ylabel=5)))</code></pre>
<div id="htmlwidget-794817da17764190ecde" style="width:912px;height:576px;" class="iplot html-widget"></div>
<script type="application/json" data-for="htmlwidget-794817da17764190ecde">{"x":{"data":{"x":[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,40,41,42,43],"y":[0.062531,0.030994,0.027963,0.034735,1.2687,0.061727,0.055596,0.042498,0.05246,0.040088,0.039395,0.040144,0.060149,0.12055,0.056303,0.033247,0.025667,0.058785,0.028706,0.089683,0.032562,0.13957,0.044821,0.033381,0.35143,0.11624,0.11946,0.089435,0.17992,0.099406,0.23206,0.76698,0.57471,0.47906,0.11205,0.19347,19.612,14.516,0.15064,0.11459,0.031075,0.1222,0.092702],"group":[1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1,1],"indID":["X-5187-20 (4.4%)","X-5467-20 (3.6%)","X-8542-20 (3.6%)","X-8638-20 (3.5%)","X-5190-20 (32.8%)","X-5490-20 (4.4%)","X-8544-20 (4.3%)","X-5195-20 (4.0%)","X-5499-20 (4.1%)","X-8560-20 (3.7%)","X-5198-20 (3.8%)","X-5507-20 (3.8%)","X-8572-20 (7.9%)","X-5232-20 (7.5%)","X-5509-20 (4.2%)","X-8575-20 (3.6%)","X-5237-20 (3.5%)","X-5548-20 (4.5%)","X-8577-20 (3.5%)","X-2173-20 (4.9%)","X-5244-20 (3.8%)","X-8294-20 (5.5%)","X-8578-20 (4.1%)","X-2178-20 (3.9%)","X-5443-20 (15.0%)","X-8296-20 (5.3%)","X-8585-20 (5.4%)","X-2198-20 (4.7%)","X-5448-20 (6.0%)","X-8303-20 (4.7%)","X-8586-20 (6.7%)","X-5150-20 (24.5%)","X-5455-20 (10.7%)","X-8306-20 (8.7%)","X-8610-20 (5.0%)","X-5169-20 (6.2%)","B6-1622 (36.8%)","X-8503-20 (34.1%)","X-8614-20 (5.2%)","X-5181-20 (5.2%)","X-5466-20 (3.8%)","X-8528-20 (5.2%)","X-8637-20 (4.9%)"]},"chartOpts":{"xlab":"Mouse","ylab":"Percent genotyping errors","ylim":[0,15],"axispos":{"xtitle":25,"ytitle":50,"xlabel":5,"ylabel":5}}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>save(errors_ind, file = &quot;data/errors_ind_wcont_redone.RData&quot;)</code></pre>
</div>
<div id="apparent-genotyping-errors" class="section level3">
<h3>Apparent Genotyping Errors</h3>
<pre class="r"><code>#load(&quot;data/snpg_wcont_redone.RData&quot;)
#snpg &lt;- gm$geno

snpgn &lt;- snpg[rownames(gm$covar),]
#snpgn &lt;- do.call(&quot;cbind&quot;, gm$geno)

gobs &lt;- do.call(&quot;cbind&quot;, gm$geno)
gobs[gobs==0] &lt;- NA

par(pty=&quot;s&quot;)
err_direct &lt;- rowMeans(snpgn != gobs, na.rm=TRUE)*100
errors_ind_0 &lt;- rowSums(e &gt; 0)[rownames(gm$covar)]/n_typed(gm)*100
par(mar=c(4.1,4.1,0.6, 0.6))
grayplot(errors_ind_0, err_direct,
         xlab=&quot;Percent errors (error LOD &gt; 0)&quot;,
         ylab=&quot;Percent errors (obs vs predicted)&quot;,
         xlim=c(0, 35), ylim=c(0, 35))
abline(0,1,lty=2, col=&quot;gray60&quot;)</code></pre>
<p><img src="figure/2.1_sample_bqc_wcont_redone.Rmd/apparent%20genotyping%20errors-1.png" width="912" style="display: block; margin: auto;" /></p>
<pre class="r"><code>pdf(file = &quot;output/Percent_genotype_errors_obs_vs_predicted_wcont_redone.pdf&quot;,width = 20, height = 20) 
par(pty=&quot;s&quot;)
err_direct &lt;- rowMeans(snpgn != gobs, na.rm=TRUE)*100
errors_ind_0 &lt;- rowSums(e &gt; 0)[rownames(gm$covar)]/n_typed(gm)*100
par(mar=c(4.1,4.1,0.6, 0.6))
grayplot(errors_ind_0, err_direct,
         xlab=&quot;Percent errors (error LOD &gt; 0)&quot;,
         ylab=&quot;Percent errors (obs vs predicted)&quot;,
         xlim=c(0, 35), ylim=c(0, 35))
abline(0,1,lty=2, col=&quot;gray60&quot;)
dev.off()</code></pre>
<pre><code>quartz_off_screen 
                2 </code></pre>
<pre class="r"><code>par(pty=&quot;s&quot;)
par(mar=c(4.1,4.1,0.6, 0.6))
grayplot(errors_ind_0, err_direct,
         xlab=&quot;Percent errors (error LOD &gt; 0)&quot;,
         ylab=&quot;Percent errors (obs vs predicted)&quot;,
         xlim=c(0, 35), ylim=c(0, 35))
abline(0,1,lty=2, col=&quot;gray60&quot;)</code></pre>
<p><img src="figure/2.1_sample_bqc_wcont_redone.Rmd/apparent%20genotyping%20errors-2.png" width="912" style="display: block; margin: auto;" /></p>
</div>
<div id="removing-samples" class="section level3">
<h3>Removing Samples</h3>
<pre class="r"><code>##percent missing
gm.covar = data.frame(id=as.character(do.call(rbind.data.frame, strsplit(rownames(gm$covar), &quot;_&quot;))[,6]),gm$covar)
qc_info &lt;- merge(gm.covar,
                  data.frame(id = names(percent_missing),percent_missing = percent_missing,stringsAsFactors = F),by = &quot;id&quot;)

#cross_over
qc_info &lt;- merge(qc_info,
                 data.frame(id = rownames(cross_over),
                            Number_crossovers = cross_over$Number_crossovers,stringsAsFactors = F),by = &quot;id&quot;)

#missing sex
#qc_info$sex.match &lt;- ifelse(qc_info$sexp == qc_info$sex, TRUE, FALSE)
rownames(samples) &lt;- as.character(samples$Original.Mouse.ID)
samples &lt;- samples[as.character(qc_info$id),]
#samples$Unique.Sample.ID &lt;- as.character(samples$Unique.Sample.ID)
all.equal(as.character(qc_info$id), as.character(samples$Original.Mouse.ID))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>qc_info$sex.match &lt;- ifelse(samples$Inferred.Sex == substring(samples$Sex, 1, 1), TRUE, FALSE)


#genotype errors
qc_info &lt;- merge(qc_info,
                 data.frame(id = as.character(do.call(rbind.data.frame, strsplit(names(errors_ind), &quot;_&quot;))[,6]),
                            genotype_erros = errors_ind,stringsAsFactors = F),by = &quot;id&quot;)

##duplicated id to be remove
qc_info$duplicate.id &lt;- ifelse(qc_info$id %in% as.character(do.call(rbind.data.frame, strsplit(summary.cg$remove.id, &quot;_&quot;))[,6]), TRUE,FALSE)

#bad.sample &lt;- qc_info[qc_info$generation ==1 | qc_info$Number_crossovers &lt;= 200 | qc_info$Number_crossovers &gt;=1000 | qc_info$percent_missing &gt;= 10 | qc_info$genotype_erros &gt;= 1 | qc_info$remove.id.duplicated == TRUE,]
bad.sample &lt;- qc_info[qc_info$generation ==1 | qc_info$Number_crossovers &lt;= 200 | qc_info$Number_crossovers &gt;=1000 | qc_info$percent_missing &gt;= 10 | qc_info$genotype_erros &gt;= 1 | qc_info$duplicate.id == TRUE,]

save(qc_info, bad.sample, file = &quot;data/qc_info_bad_sample_wcont_redone.RData&quot;)

gm_samqc &lt;- gm[paste0(&quot;-&quot;,as.character(bad.sample$id.1)),]

#fix sex mismatch
#gm_samqc$covar[gm_samqc$covar$id %in% names(yint_ave[!names(yint_ave) %in% bad.sample$id][yint_ave[!names(yint_ave) %in% bad.sample$id] &lt;= 0.1]),&quot;sexp&quot;] &lt;- &quot;F&quot;

gm_samqc</code></pre>
<pre><code>Object of class cross2 (crosstype &quot;do&quot;)

Total individuals                37
No. genotyped individuals        37
No. phenotyped individuals       37
No. with both geno &amp; pheno       37

No. phenotypes                    1
No. covariates                    3
No. phenotype covariates          0

No. chromosomes                  20
Total markers                137220

No. markers by chr:
    1     2     3     4     5     6     7     8     9    10    11    12    13 
10415 10432  8203  7951  8028  8125  7756  6714  6979  6628  7430  6441  6323 
   14    15    16    17    18    19     X 
 6227  5531  5174  5320  4784  3676  5083 </code></pre>
<pre class="r"><code>save(gm_samqc, file = &quot;data/gm_samqc_wcont_redone.RData&quot;)

# update other stuff
e &lt;- e[ind_ids(gm_samqc),]
g &lt;- g[ind_ids(gm_samqc),]
snpg &lt;- snpg[ind_ids(gm_samqc),]

save(e,g,snpg, file = &quot;data/e_g_snpg_samqc_wcont_redone.RData&quot;)
#save(e,g, file = &quot;data/e_g_snpg_samqc_wcont_redone.RData&quot;)</code></pre>
<p>Here is the list of samples that were removed:</p>
<table class="table table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto; width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;">
Genoprobs_Sample_ID
</th>
<th style="text-align:center;">
Neogen_Sample_ID
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;">
Jackson_Lab_Korstanje_MURGIGV01_20210604_B6-1622_B12
</td>
<td style="text-align:center;">
B6-1622
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-5150-20_E11
</td>
<td style="text-align:center;">
X-5150-20
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_Korstanje_MURGIGV01_20210701_X-5190-20_H5
</td>
<td style="text-align:center;">
X-5190-20
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-5443-20_F10
</td>
<td style="text-align:center;">
X-5443-20
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-5455-20_F11
</td>
<td style="text-align:center;">
X-5455-20
</td>
</tr>
<tr>
<td style="text-align:center;">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-8503-20_C12
</td>
<td style="text-align:center;">
X-8503-20
</td>
</tr>
</tbody>
</table>
<p>Below is a table summarising the problematic samples found throughout QC. These include the following:</p>
<ul>
<li>no_pheno == sample does not have a phenotype (if applicable)</li>
<li>high_miss == sample has higher than 10% missing genotypes</li>
<li>diff_sex == sample has sex mismatch</li>
<li>high_xo == sample had crossover events greater than 1000 and less than 200</li>
<li>high_geno_errors == Sample has geno errors above 1%</li>
<li>DO_generation == DO generation is 1</li>
<li>dublicate_ID == sample was duplicated</li>
</ul>
<p><strong>NB:</strong> For duplcate pairs, the one that was chosen to be removed was the one that had a higher missing rate</p>
<table class="table table-striped" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;">
Genoprobs_Sample_ID
</th>
<th style="text-align:center;">
Neogen_Sample_ID
</th>
<th style="text-align:center;">
high_miss
</th>
<th style="text-align:center;">
diff_sex
</th>
<th style="text-align:center;">
high_xo
</th>
<th style="text-align:center;">
DO.Generation
</th>
<th style="text-align:center;">
high_geno.errors
</th>
<th style="text-align:center;">
duplicate_ID
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210604_B6-1622_B12
</td>
<td style="text-align:center;">
B6-1622
</td>
<td style="text-align:center;">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
<span style="     color: blue !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: blue !important;">XX</span>
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
<span style="     color: lightgreen !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: lightgreen !important;">XX</span>
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-5150-20_E11
</td>
<td style="text-align:center;">
X-5150-20
</td>
<td style="text-align:center;">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210701_X-5190-20_H5
</td>
<td style="text-align:center;">
X-5190-20
</td>
<td style="text-align:center;">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-5443-20_F10
</td>
<td style="text-align:center;">
X-5443-20
</td>
<td style="text-align:center;">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-5455-20_F11
</td>
<td style="text-align:center;">
X-5455-20
</td>
<td style="text-align:center;">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cm; ">
Jackson_Lab_Korstanje_MURGIGV01_20210604_X-8503-20_C12
</td>
<td style="text-align:center;">
X-8503-20
</td>
<td style="text-align:center;">
<span style="     color: red !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: red !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: purple !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: purple !important;">XX</span>
</td>
<td style="text-align:center;">
</td>
<td style="text-align:center;">
<span style="     color: pink !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: pink !important;">XX</span>
</td>
<td style="text-align:center;">
<span style="     color: orange !important;border-radius: 4px; padding-right: 4px; padding-left: 4px; background-color: orange !important;">XX</span>
</td>
</tr>
</tbody>
</table>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readxl_1.3.1      cluster_2.1.0     dplyr_0.8.5       optparse_1.6.6   
 [5] rhdf5_2.28.1      tidyr_1.0.2       data.table_1.12.8 fst_0.9.2        
 [9] knitr_1.28        kableExtra_1.1.0  mclust_5.4.6      ggrepel_0.8.2    
[13] ggplot2_3.3.2     qtlcharts_0.11-6  qtl2_0.22         broman_0.70-4    
[17] workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6      assertthat_0.2.1  rprojroot_1.3-2   digest_0.6.25    
 [5] R6_2.4.1          cellranger_1.1.0  backports_1.1.7   RSQLite_2.2.0    
 [9] evaluate_0.14     httr_1.4.1        highr_0.8         pillar_1.4.4     
[13] rlang_0.4.6       rstudioapi_0.11   whisker_0.4       blob_1.2.1       
[17] rmarkdown_2.1     labeling_0.3      qtl_1.46-2        webshot_0.5.2    
[21] readr_1.3.1       stringr_1.4.0     htmlwidgets_1.5.1 bit_4.0.4        
[25] munsell_0.5.0     compiler_3.6.2    httpuv_1.5.2      xfun_0.13        
[29] pkgconfig_2.0.3   htmltools_0.4.0   tidyselect_1.0.0  tibble_3.0.1     
[33] viridisLite_0.3.0 crayon_1.3.4      withr_2.2.0       later_1.0.0      
[37] grid_3.6.2        jsonlite_1.6.1    gtable_0.3.0      lifecycle_0.2.0  
[41] DBI_1.1.0         git2r_0.26.1      magrittr_1.5      scales_1.1.1     
[45] stringi_1.4.6     farver_2.0.3      fs_1.4.1          promises_1.1.0   
[49] getopt_1.20.3     xml2_1.3.1        ellipsis_0.3.1    vctrs_0.3.1      
[53] Rhdf5lib_1.6.3    tools_3.6.2       bit64_0.9-7       glue_1.4.1       
[57] purrr_0.3.4       hms_0.5.3         parallel_3.6.2    yaml_2.2.1       
[61] colorspace_1.4-1  rvest_0.3.5       memoise_1.1.0    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
