<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Belinda Cornes" />

<meta name="date" content="2021-03-25" />

<title>Genotype Concordance [HaploQA vs. R/qtl2]</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/spacelab.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Korstanje-OConnor Workflow</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/TheJacksonLaboratory/Korstanje-OConnor_Workflow">
    <span class="fa fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Genotype Concordance [HaploQA vs. R/qtl2]</h1>
<h4 class="author">Belinda Cornes</h4>
<h4 class="date">2021-03-25</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#loading-project">Loading Project</a></li>
<li><a href="#concordance-snp-level-genotypes">Concordance (SNP-level genotypes)</a></li>
<li><a href="#concordance-phased-haplotypes-genotypes">Concordance (phased haplotypes genotypes)</a></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2021-03-25
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Korstanje-OConnor_Workflow/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20210317code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20210317)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20210317code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20210317)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomTheJacksonLaboratoryKorstanjeOConnorWorkflowtreed7dde7b43b071cb481f372afc59815cae25ae503targetblankd7dde7ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/TheJacksonLaboratory/Korstanje-OConnor_Workflow/tree/d7dde7b43b071cb481f372afc59815cae25ae503" target="_blank">d7dde7b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomTheJacksonLaboratoryKorstanjeOConnorWorkflowtreed7dde7b43b071cb481f372afc59815cae25ae503targetblankd7dde7ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/TheJacksonLaboratory/Korstanje-OConnor_Workflow/tree/d7dde7b43b071cb481f372afc59815cae25ae503" target="_blank">d7dde7b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    analysis/.DS_Store

Untracked files:
    Untracked:  data/bad_markers_all.RData
    Untracked:  data/bad_markers_all_wcont.RData
    Untracked:  data/e.RData
    Untracked:  data/e_g_snpg_samqc.RData
    Untracked:  data/e_g_snpg_samqc_wcont.RData
    Untracked:  data/e_wcont.RData
    Untracked:  data/errors_ind.RData
    Untracked:  data/errors_ind_wcont.RData
    Untracked:  data/genotype_errors_marker.RData
    Untracked:  data/genotype_errors_marker_wcont.RData
    Untracked:  data/genotype_freq_marker.RData
    Untracked:  data/genotype_freq_marker_wcont.RData
    Untracked:  data/gm_allqc.RData
    Untracked:  data/gm_allqc_wcont.RData
    Untracked:  data/gm_samqc.RData
    Untracked:  data/gm_samqc_wcont.RData
    Untracked:  data/m.RData
    Untracked:  data/m_wcont.RData
    Untracked:  data/nxo.RData
    Untracked:  data/nxo_wcont.RData
    Untracked:  data/percent_missing_id.RData
    Untracked:  data/percent_missing_id_wcont.RData
    Untracked:  data/percent_missing_marker.RData
    Untracked:  data/percent_missing_marker_wcont.RData
    Untracked:  data/qc_info_bad_sample.RData
    Untracked:  data/qc_info_bad_sample_wcont.RData
    Untracked:  data/snpg.RData
    Untracked:  data/snpg_wcont.RData
    Untracked:  data/summary.cg.RData
    Untracked:  data/summary.cg_wcont.RData
    Untracked:  output/Percent_genotype_errors_obs_vs_predicted.pdf
    Untracked:  output/Percent_genotype_errors_obs_vs_predicted_wcont.pdf
    Untracked:  output/Percent_missing_genotype_data.pdf
    Untracked:  output/Percent_missing_genotype_data_per_marker.pdf
    Untracked:  output/Percent_missing_genotype_data_per_marker_wcont.pdf
    Untracked:  output/Percent_missing_genotype_data_wcont.pdf
    Untracked:  output/genotype_error_marker.pdf
    Untracked:  output/genotype_error_marker_wcont.pdf
    Untracked:  output/genotype_frequency_marker.pdf
    Untracked:  output/genotype_frequency_marker_wcont.pdf
    Untracked:  output/number_crossover.pdf
    Untracked:  output/number_crossover_wcont.pdf

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/concordance_random.samples.Rmd</code>) and HTML (<code>docs/concordance_random.samples.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/TheJacksonLaboratory/Korstanje-OConnor_Workflow/blob/d7dde7b43b071cb481f372afc59815cae25ae503/analysis/concordance_random.samples.Rmd" target="_blank">d7dde7b</a>
</td>
<td>
Belinda Cornes
</td>
<td>
2021-03-25
</td>
<td>
concordance update
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="loading-project" class="section level3">
<h3>Loading Project</h3>
<pre class="r"><code>#selected mice to check concordance
concord.mice &lt;- c(&quot;B6 - 1616A&quot;, 
                &quot;B6 - 1616&quot;, 
                &quot;B6 - 1618A&quot;, 
                &quot;B6 - 1618&quot;, 
                &quot;A-1620-20A&quot;, 
                &quot;A-1620-20B&quot;, 
                &quot;A-1620-20C&quot;, 
                &quot;A-1620-20&quot;, 
                &quot;A-1626-20D&quot;, 
                &quot;A-1626-20&quot;,
                &quot;X-8503-20&quot;,
                &quot;X-5185-20&quot;,
                &quot;X-5186-20&quot;,
                &quot;X-8612-20&quot;,
                &quot;X-8315-20&quot;,
                &quot;5257&quot;,
                &quot;X-2199-20&quot;,
                &quot;X-5172-20&quot;,
                &quot;X-2207-20&quot;
                )

filepaths = &quot;/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/genoprobs/qced.samples/&quot;

############################

##extracting number genotypes from r/qtl2 data for selected samples
gmf &lt;- read_cross2(&quot;/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/genoprobs/all.samples/qtl2/gm_sorted.json&quot;)
gmf</code></pre>
<pre><code>Object of class cross2 (crosstype &quot;do&quot;)

Total individuals               474
No. genotyped individuals       474
No. phenotyped individuals      474
No. with both geno &amp; pheno      474

No. phenotypes                    1
No. covariates                    3
No. phenotype covariates          0

No. chromosomes                  20
Total markers                137220

No. markers by chr:
    1     2     3     4     5     6     7     8     9    10    11    12    13 
10415 10432  8203  7951  8028  8125  7756  6714  6979  6628  7430  6441  6323 
   14    15    16    17    18    19     X 
 6227  5531  5174  5320  4784  3676  5083 </code></pre>
<pre class="r"><code>gm &lt;- gmf[ind_ids(gmf) %in%concord.mice,]
gm</code></pre>
<pre><code>Object of class cross2 (crosstype &quot;do&quot;)

Total individuals                19
No. genotyped individuals        19
No. phenotyped individuals       19
No. with both geno &amp; pheno       19

No. phenotypes                    1
No. covariates                    3
No. phenotype covariates          0

No. chromosomes                  20
Total markers                137220

No. markers by chr:
    1     2     3     4     5     6     7     8     9    10    11    12    13 
10415 10432  8203  7951  8028  8125  7756  6714  6979  6628  7430  6441  6323 
   14    15    16    17    18    19     X 
 6227  5531  5174  5320  4784  3676  5083 </code></pre>
<pre class="r"><code>rqtl.genos.numbers &lt;- t(as.data.frame(do.call(cbind,gm$geno)))
rqtl.genos.numbers &lt;- as.data.frame(rqtl.genos.numbers)

sample_file &lt;- dir(path = filepaths, pattern = &quot;^DODB_*&quot;, full.names = TRUE)
samples.all &lt;- read.csv(sample_file)
samples &lt;- samples.all[samples.all$Original.Mouse.ID %in%concord.mice,]
all.equal(as.character(ind_ids(gm)), as.character(samples$Unique.Sample.ID))</code></pre>
<pre><code>[1] TRUE</code></pre>
<pre class="r"><code>############################

##extracting letter genotypes from raw data for selected samples
#combining genotype files:
main.path = &quot;/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/Neogen/&quot;
path.dirs &lt;- dir(path = main.path, pattern = &quot;^Jackson_Lab_Korstanje_MURGIGV01_*&quot;, full.names = TRUE)

geno_files &lt;- list()
for (i in (1:length(path.dirs))){
    
    ##final report level data
    #filepathg = &#39;/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/Neogen/Jackson_Lab_Korstanje_MURGIGV01_20210211/&#39;
    #filepaths = &#39;/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/genoprobs/qced.samples/&#39;
    #filepathp = &#39;/projects/compsci/corneb_cs/workflowr/Thaiss/Thaiss_workflowr_2/data/&#39;

    report_file &lt;- dir(path = path.dirs[i], pattern = &quot;_FinalReport.zip$&quot;, full.names = TRUE)
    sample_file &lt;- dir(path = filepaths, pattern = &quot;^DODB_*&quot;, full.names = TRUE)
    #pheno_file &lt;- dir(path = filepathp, pattern = &quot;Uni_of_Penn_Thaiss_pheno.3.csv&quot;, full.names = TRUE)

    report_file_hdr &lt;- unlist(strsplit(report_file, split = &quot;/&quot;))
    report_file_hdr &lt;- report_file_hdr[length(report_file_hdr)]

    header &lt;- read.table(unz(report_file, sub(pattern = &quot;zip$&quot;, replacement = &quot;txt&quot;, report_file_hdr)), 
                          sep = &quot;\t&quot;, skip = 1, nrows = 2)
    report &lt;- fread(paste(&quot;unzip -cq&quot;, report_file),
                     skip = 9, showProgress = TRUE, sep = &quot;\t&quot;)
    #samples &lt;- fread(paste0(&quot;unzip -cq &quot;,  sample_file))
    colnames(report) &lt;- gsub(&quot; |-&quot;, &quot;_&quot;, colnames(report))

    geno &lt;- report %&gt;%
        select(Sample_ID, SNP_Name, Allele1___Forward, Allele2___Forward) %&gt;%
        mutate(Sample_ID = factor(Sample_ID, levels = unique(report$Sample_ID))) %&gt;%
        unite(genotype, c(&quot;Allele1___Forward&quot;, &quot;Allele2___Forward&quot;), sep = &quot;&quot;) %&gt;%
        spread(Sample_ID, genotype)

    geno_files[[i]] &lt;- geno

}</code></pre>
<pre><code>Taking input= as a system command (&#39;unzip -cq /Users/corneb/Documents/MyJax/CS/Projects/Oconnor/Neogen//Jackson_Lab_Korstanje_MURGIGV01_20201111/Jackson_Lab_Korstanje_MURGIGV01_20201111_FinalReport.zip&#39;) and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
<pre><code>Taking input= as a system command (&#39;unzip -cq /Users/corneb/Documents/MyJax/CS/Projects/Oconnor/Neogen//Jackson_Lab_Korstanje_MURGIGV01_20210211/Jackson_Lab_Korstanje_MURGIGV01_20210211_FinalReport.zip&#39;) and a variable has been used in the expression passed to `input=`. Please use fread(cmd=...). There is a security concern if you are creating an app, and the app could have a malicious user, and the app is not running in a secure environment; e.g. the app is running as root. Please read item 5 in the NEWS file for v1.11.6 for more information and for the option to suppress this message.</code></pre>
<pre class="r"><code>all_genos &lt;- Reduce(function(x, y) merge(x, y,by=c(&quot;SNP_Name&quot;),all=TRUE),geno_files)
rownames(all_genos) &lt;- all_genos$SNP_Name
all_genos_edit &lt;- all_genos[c(2:length(all_genos))]

all_genos_t &lt;- as.data.frame(t(all_genos_edit))
all_genos_t$sample.ids &lt;- rownames(all_genos_t)

rqtl.genos.t &lt;- all_genos_t[all_genos_t$sample.ids %in%concord.mice,]
rqtl.genos &lt;- all_genos_edit

############################

##Pulling in haploQA snp level data for selected samples

main.path.2 = &quot;/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/haploQA/&quot;
files &lt;- dir(path = main.path.2, pattern = &quot;_snp_report.txt$&quot;, full.names = TRUE)

haploqa_files &lt;- list()
for (i in (1:length(files))){

    file &lt;- read.table(files[i], header=T, sep=&quot;\t&quot;)
    #sample.id &lt;- unique(file$original_sample_id)
    rownames(file) &lt;- file$snp_id
    file$genotype &lt;- paste0(file$allele1_fwd, file$allele2_fwd)
    names(file)[10] &lt;-c(unique(as.character(file$original_sample_id)))
    file.new &lt;- file[c(3,10)]

    haploqa_files[[i]] &lt;- file.new

}

all_haploqa_files &lt;- Reduce(function(x, y) merge(x, y,by=c(&quot;snp_id&quot;),all=TRUE),haploqa_files)
rownames(all_haploqa_files) &lt;- all_haploqa_files$snp_id
all_haploqa_files_edit &lt;- all_haploqa_files[c(2:length(all_haploqa_files))]

haploqa.genos.t &lt;- as.data.frame(t(all_haploqa_files_edit))
haploqa.genos.t$sample.ids &lt;- rownames(haploqa.genos.t)
haploqa.genos &lt;- all_haploqa_files_edit</code></pre>
</div>
<div id="concordance-snp-level-genotypes" class="section level3">
<h3>Concordance (SNP-level genotypes)</h3>
<pre class="r"><code>#merging two data frames together: genos &amp; haploqa on snp.id

all_genos.new &lt;- all_genos[,colnames(all_genos) %in%concord.mice]
all_genos.new$SNP_Name &lt;- rownames(all_genos.new)
geno.letters &lt;- merge(all_genos.new,all_haploqa_files, by.x=c(&quot;SNP_Name&quot;), by.y=c(&quot;snp_id&quot;), all=T)

concord.mice.2 = set_names(colnames(geno.letters[2:length(geno.letters)]))

concordance_letters_hm = function(x, y) {

    #sample.match &lt;- ifelse(geno.letters[[&quot;B6 - 1616A&quot;]] == geno.letters[[&quot;B6 - 1616&quot;]], 1, 0)
    sample.match &lt;- ifelse(geno.letters[[x]] == geno.letters[[y]], 1, 0)
    concordance.id1 &lt;- x
    concordance.id2 &lt;- y
    concordance.sum &lt;- sum(sample.match)
    concordance.percentage &lt;- (concordance.sum/nrow(geno.letters))*100
    concordance.letters &lt;- cbind(concordance.id1,concordance.id2,concordance.percentage)
    write.table(concordance.letters,paste0(&quot;/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/concordance/concordance.letters_&quot;,x,&quot;_vs_&quot;,y,&quot;.txt&quot;), quote=F, sep=&quot;\t&quot;)

}

all.pairs &lt;- map(concord.mice.2, ~map(concord.mice.2, concordance_letters_hm, y = .x))

main.path.3 = &quot;/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/concordance/&quot;
files &lt;- dir(path = main.path.3, pattern = &quot;.txt$&quot;, full.names = TRUE)

concordance_letters &lt;- list()
for (i in (1:length(files))){
    #sample.id &lt;- paste0(&quot;^&quot;,i,&quot;[.]&quot;)
    file &lt;- read.table(files[i], header=T, sep=&quot;\t&quot;)
    concordance_letters[[i]] &lt;- file
}

concordance &lt;- as.data.frame(do.call(rbind, concordance_letters))

#removing those lines where comparing with same platform
concordance.subset.1 &lt;- subset(concordance,  sapply(strsplit(concordance$concordance.id1,&#39;.&#39;,fixed=T),function(x) x[2]) != sapply(strsplit(concordance$concordance.id2,&#39;.&#39;,fixed=T),function(x) x[2]))
concordance.subset.2 &lt;- subset(concordance.subset.1,  sapply(strsplit(concordance.subset.1$concordance.id1,&#39;.&#39;,fixed=T),function(x) x[2]) != &#39;y&#39;)
concordance.subset.2 &lt;- as.data.frame(concordance.subset.2)

#Removing .x and .y
concordance.subset.2$concordance.id1 &lt;- gsub(&quot;.x&quot;, &quot;&quot;, concordance.subset.2$concordance.id1)
concordance.subset.2$concordance.id2 &lt;- gsub(&quot;.y&quot;, &quot;&quot;, concordance.subset.2$concordance.id2)

#Rmoving same pairs

#names(concordance) &lt;- c(&quot;concordance [snp-level genotypes]&quot;)
#concordance$Neogen_Sample_ID &lt;- rownames(concordance)
#concordance$Neogen_Sample_ID_2 &lt;- rownames(concordance)

##Keeping pairs of same IDs
concordance.pairs &lt;- subset(concordance.subset.2,  concordance.subset.2$concordance.id1 == concordance.subset.2$concordance.id2)
names(concordance.pairs) &lt;- c(&quot;R/qtl2 Data&quot;,&quot;haploQA Data&quot;,&quot;Concordance Percentage&quot;)
concordance.pairs$&quot;Concordance Percentage&quot; &lt;- paste0(concordance.pairs$&quot;Concordance Percentage&quot;,&quot;%&quot;)

rownames(concordance.pairs) &lt;- NULL

concordance.pairs[] %&gt;%
  kable(escape = F,align = c(&quot;ccccccccccccccccc&quot;),linesep =&quot;\\hline&quot;) %&gt;%
  kable_styling(&quot;striped&quot;, full_width = T) %&gt;%
  column_spec(1:2, width = &quot;5cms&quot;) </code></pre>
<table class="table table-striped" style="margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:center;">
R/qtl2 Data
</th>
<th style="text-align:center;">
haploQA Data
</th>
<th style="text-align:center;">
Concordance Percentage
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:center;width: 5cms; ">
5257
</td>
<td style="text-align:center;width: 5cms; ">
5257
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
A-1620-20
</td>
<td style="text-align:center;width: 5cms; ">
A-1620-20
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
A-1620-20A
</td>
<td style="text-align:center;width: 5cms; ">
A-1620-20A
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
A-1620-20B
</td>
<td style="text-align:center;width: 5cms; ">
A-1620-20B
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
A-1620-20C
</td>
<td style="text-align:center;width: 5cms; ">
A-1620-20C
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
A-1626-20
</td>
<td style="text-align:center;width: 5cms; ">
A-1626-20
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
A-1626-20D
</td>
<td style="text-align:center;width: 5cms; ">
A-1626-20D
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
B6 - 1616
</td>
<td style="text-align:center;width: 5cms; ">
B6 - 1616
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
B6 - 1616A
</td>
<td style="text-align:center;width: 5cms; ">
B6 - 1616A
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
B6 - 1618
</td>
<td style="text-align:center;width: 5cms; ">
B6 - 1618
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
B6 - 1618A
</td>
<td style="text-align:center;width: 5cms; ">
B6 - 1618A
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
X-2199-20
</td>
<td style="text-align:center;width: 5cms; ">
X-2199-20
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
X-2207-20
</td>
<td style="text-align:center;width: 5cms; ">
X-2207-20
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
X-5172-20
</td>
<td style="text-align:center;width: 5cms; ">
X-5172-20
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
X-5185-20
</td>
<td style="text-align:center;width: 5cms; ">
X-5185-20
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
X-5186-20
</td>
<td style="text-align:center;width: 5cms; ">
X-5186-20
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
X-8315-20
</td>
<td style="text-align:center;width: 5cms; ">
X-8315-20
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
X-8503-20
</td>
<td style="text-align:center;width: 5cms; ">
X-8503-20
</td>
<td style="text-align:center;">
100%
</td>
</tr>
<tr>
<td style="text-align:center;width: 5cms; ">
X-8612-20
</td>
<td style="text-align:center;width: 5cms; ">
X-8612-20
</td>
<td style="text-align:center;">
100%
</td>
</tr>
</tbody>
</table>
<pre class="r"><code>##Image of all sample IDs compared to each other (from two different paltforms)
image = (p &lt;- ggplot(concordance.subset.2, aes(concordance.id1, concordance.id2)) + 
geom_tile(aes(fill = concordance.percentage)) + 
#geom_text(aes(fill = concordance.percentage, label = concordance.percentage), size=0.1) +
scale_fill_gradient2(low = &quot;blue&quot;, high = &quot;red&quot;) +
xlab(&quot;R/qtl2 Data&quot;) +
ylab(&quot;haploQA Data&quot;) +
ggtitle(&quot;Concordance between samples [R/qtl2 vs. haploQA data]&quot;) + 
theme(axis.ticks = element_line(size = 0.01),
      axis.text.x =  element_text(angle=90),
      axis.title.x = element_text(vjust=-2),
      axis.title.y = element_text(vjust=2.3),
      #axis.text.y =  element_text(size=8),
      plot.title = element_text(vjust=2),
      legend.position = &quot;right&quot;,
      plot.margin = unit(c(1,1,1,1), &quot;cm&quot;)))

image</code></pre>
<p><img src="figure/concordance_random.samples.Rmd/concordance%20letters-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##Image of all sample IDs compared to each other (R/qtl2)
concordance.subset.1 &lt;- subset(concordance,  sapply(strsplit(concordance$concordance.id1,&#39;.&#39;,fixed=T),function(x) x[2]) == sapply(strsplit(concordance$concordance.id2,&#39;.&#39;,fixed=T),function(x) x[2]))
concordance.subset.2 &lt;- subset(concordance.subset.1,  sapply(strsplit(concordance.subset.1$concordance.id1,&#39;.&#39;,fixed=T),function(x) x[2]) != &#39;y&#39;)
concordance.subset.2 &lt;- as.data.frame(concordance.subset.2)

#Removing .x and .y
concordance.subset.2$concordance.id1 &lt;- gsub(&quot;.x&quot;, &quot;&quot;, concordance.subset.2$concordance.id1)
concordance.subset.2$concordance.id2 &lt;- gsub(&quot;.x&quot;, &quot;&quot;, concordance.subset.2$concordance.id2)

image = (p &lt;- ggplot(concordance.subset.2, aes(concordance.id1, concordance.id2)) + 
geom_tile(aes(fill = concordance.percentage)) + 
#geom_text(aes(fill = concordance.percentage, label = concordance.percentage), size=0.1) +
scale_fill_gradient2(low = &quot;blue&quot;, high = &quot;red&quot;) +
xlab(&quot;Individual 1&quot;) +
ylab(&quot;Individual 2&quot;) +
ggtitle(&quot;Concordance between samples [R/qtl2 data]&quot;) + 
theme(axis.ticks = element_line(size = 0.01),
      axis.text.x =  element_text(angle=90),
      axis.title.x = element_text(vjust=-2),
      axis.title.y = element_text(vjust=2.3),
      #axis.text.y =  element_text(size=8),
      plot.title = element_text(vjust=2),
      legend.position = &quot;right&quot;,
      plot.margin = unit(c(1,1,1,1), &quot;cm&quot;)))

image</code></pre>
<p><img src="figure/concordance_random.samples.Rmd/concordance%20letters-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>##Image of all sample IDs compared to each other (haploQA)
concordance.subset.2 &lt;- subset(concordance.subset.1,  sapply(strsplit(concordance.subset.1$concordance.id1,&#39;.&#39;,fixed=T),function(x) x[2]) != &#39;x&#39;)
concordance.subset.2 &lt;- as.data.frame(concordance.subset.2)

#Removing .x and .y
concordance.subset.2$concordance.id1 &lt;- gsub(&quot;.y&quot;, &quot;&quot;, concordance.subset.2$concordance.id1)
concordance.subset.2$concordance.id2 &lt;- gsub(&quot;.y&quot;, &quot;&quot;, concordance.subset.2$concordance.id2)

image = (p &lt;- ggplot(concordance.subset.2, aes(concordance.id1, concordance.id2)) + 
geom_tile(aes(fill = concordance.percentage)) + 
#geom_text(aes(fill = concordance.percentage, label = concordance.percentage), size=0.1) +
scale_fill_gradient2(low = &quot;blue&quot;, high = &quot;red&quot;) +
xlab(&quot;Individual 1&quot;) +
ylab(&quot;Individual 2&quot;) +
ggtitle(&quot;Concordance between samples [haploQA data]&quot;) + 
theme(axis.ticks = element_line(size = 0.01),
      axis.text.x =  element_text(angle=90),
      axis.title.x = element_text(vjust=-2),
      axis.title.y = element_text(vjust=2.3),
      #axis.text.y =  element_text(size=8),
      plot.title = element_text(vjust=2),
      legend.position = &quot;right&quot;,
      plot.margin = unit(c(1,1,1,1), &quot;cm&quot;)))

image</code></pre>
<p><img src="figure/concordance_random.samples.Rmd/concordance%20letters-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="concordance-phased-haplotypes-genotypes" class="section level3">
<h3>Concordance (phased haplotypes genotypes)</h3>
<pre class="r"><code>#merging two data frames together: genos &amp; haploqa on snp.id
##concordance between gm$geno (numbers) to haploqa (numbers --&gt; need to change letters to numbers)

#rqtl.genos.numbers$SNP_Name &lt;- rownames(rqtl.genos.numbers)
#rqtl.genos.numbers &lt;- rqtl.genos.numbers[order(rqtl.genos.numbers$SNP_Name),]
#haploqa.genos.numbers$snp_id &lt;- rownames(haploqa.genos.numbers)
#geno.numbers &lt;- merge(rqtl.genos.numbers,haploqa.genos.numbers, by.x=c(&quot;SNP_Name&quot;), by.y=c(&quot;snp_id&quot;), all.x=T)
#rownames(geno.numbers) &lt;- geno.numbers$SNP_Name

#concordance_numbers &lt;- list()
#for(i in concord.mice){
#   #sample.id &lt;- paste0(&quot;^&quot;,i,&quot;[.]&quot;)
#   sample.match &lt;- as.data.frame(geno.numbers[,grep(paste0(&quot;^&quot;,i,&quot;[.]&quot;),colnames(geno.numbers))])
#   print(table(sample.match[c(1)]))
#   print(table(sample.match[c(2)]))
#   sample.match$match &lt;- ifelse(sample.match[c(1)] == sample.match[c(2)], 1, 0)
#   concordance.sum &lt;- sum(sample.match$match)
#   concordance.percentage &lt;- (concordance.sum/nrow(sample.match))*100
#   concordance_numbers[[i]] &lt;- paste0(concordance.percentage,&quot;%&quot;)
#}

#concordance.no &lt;- as.data.frame(do.call(rbind, concordance_numbers))
#names(concordance.no) &lt;- c(&quot;concordance [numbers]&quot;)
#concordance.no$Neogen_Sample_ID &lt;- rownames(concordance)

#rownames(concordance.no) &lt;- NULL

#concordance.no[c(2,1)] %&gt;%
#  kable(escape = F,align = c(&quot;ccccccccccccccccc&quot;),linesep =&quot;\\hline&quot;) %&gt;%
#  kable_styling(&quot;striped&quot;, full_width = T) %&gt;%
#  column_spec(1:2, width = &quot;5cms&quot;) </code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.2 (2019-12-12)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_AU.UTF-8/en_AU.UTF-8/en_AU.UTF-8/C/en_AU.UTF-8/en_AU.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] purrr_0.3.4       ggplot2_3.3.2     qtl2_0.22         tibble_3.0.1     
 [5] readxl_1.3.1      cluster_2.1.0     dplyr_0.8.5       optparse_1.6.6   
 [9] rhdf5_2.28.1      mclust_5.4.6      tidyr_1.0.2       data.table_1.12.8
[13] knitr_1.28        kableExtra_1.1.0  workflowr_1.6.2  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4.6      assertthat_0.2.1  rprojroot_1.3-2   digest_0.6.25    
 [5] R6_2.4.1          cellranger_1.1.0  backports_1.1.7   RSQLite_2.2.0    
 [9] evaluate_0.14     httr_1.4.1        pillar_1.4.4      rlang_0.4.6      
[13] rstudioapi_0.11   whisker_0.4       blob_1.2.1        rmarkdown_2.1    
[17] labeling_0.3      webshot_0.5.2     readr_1.3.1       stringr_1.4.0    
[21] bit_1.1-15.2      munsell_0.5.0     compiler_3.6.2    httpuv_1.5.2     
[25] xfun_0.13         pkgconfig_2.0.3   htmltools_0.4.0   tidyselect_1.0.0 
[29] viridisLite_0.3.0 crayon_1.3.4      withr_2.2.0       later_1.0.0      
[33] grid_3.6.2        jsonlite_1.6.1    gtable_0.3.0      lifecycle_0.2.0  
[37] DBI_1.1.0         git2r_0.26.1      magrittr_1.5      scales_1.1.1     
[41] stringi_1.4.6     farver_2.0.3      fs_1.4.1          promises_1.1.0   
[45] getopt_1.20.3     xml2_1.3.1        ellipsis_0.3.1    vctrs_0.3.1      
[49] Rhdf5lib_1.6.3    tools_3.6.2       bit64_0.9-7       glue_1.4.1       
[53] hms_0.5.3         parallel_3.6.2    yaml_2.2.1        colorspace_1.4-1 
[57] rvest_0.3.5       memoise_1.1.0    </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>



</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
