---
title: "Genotype Concordance [HaploQA vs. R/qtl2]"
author: "Belinda Cornes"
date: "`r Sys.Date()`"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,warning=FALSE)

library("kableExtra")
library("knitr")
options(stringsAsFactors = FALSE)
library(data.table) 
library(tidyr)     
library(mclust)     
library(rhdf5)      
library(optparse)
library(dplyr)
library(cluster)
library(readxl)
library(tibble)
library(qtl2)
library(ggplot2)
library(purrr)
library(abind)
options(max.print=1000000)

```

### Loading Project
```{r extracting letter & number genos for select samples, eval=TRUE, fig.height = 6, fig.width = 9.5, fig.align = "center"}

#selected mice to check concordance
concord.mice <- c("B6 - 1616A", 
                "B6 - 1616", 
                "B6 - 1618A", 
                "B6 - 1618", 
                "A-1620-20A", 
                "A-1620-20B", 
                "A-1620-20C", 
                "A-1620-20", 
                "A-1626-20D", 
                "A-1626-20",
                "X-8503-20",
                "X-5185-20",
                "X-5186-20",
                "X-8612-20",
                "X-8315-20",
                "5257",
                "X-2199-20",
                "X-5172-20",
                "X-2207-20"
                )

filepaths = "/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/genoprobs/qced.samples/"

############################

##extracting number genotypes from r/qtl2 data for selected samples
gmf <- read_cross2("/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/genoprobs/all.samples/qtl2/gm_sorted.json")
gmf

gm <- gmf[ind_ids(gmf) %in%concord.mice,]
gm

rqtl.genos.numbers <- t(as.data.frame(do.call(cbind,gm$geno)))
rqtl.genos.numbers <- as.data.frame(rqtl.genos.numbers)

sample_file <- dir(path = filepaths, pattern = "^DODB_*", full.names = TRUE)
samples.all <- read.csv(sample_file)
samples <- samples.all[samples.all$Original.Mouse.ID %in%concord.mice,]
all.equal(as.character(ind_ids(gm)), as.character(samples$Unique.Sample.ID))

############################

##extracting letter genotypes from raw data for selected samples
#combining genotype files:
main.path = "/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/Neogen/"
path.dirs <- dir(path = main.path, pattern = "^Jackson_Lab_Korstanje_MURGIGV01_*", full.names = TRUE)

geno_files <- list()
for (i in (1:length(path.dirs))){
	
	##final report level data
	#filepathg = '/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/Neogen/Jackson_Lab_Korstanje_MURGIGV01_20210211/'
	#filepaths = '/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/genoprobs/qced.samples/'
	#filepathp = '/projects/compsci/corneb_cs/workflowr/Thaiss/Thaiss_workflowr_2/data/'

	report_file <- dir(path = path.dirs[i], pattern = "_FinalReport.zip$", full.names = TRUE)
	sample_file <- dir(path = filepaths, pattern = "^DODB_*", full.names = TRUE)
	#pheno_file <- dir(path = filepathp, pattern = "Uni_of_Penn_Thaiss_pheno.3.csv", full.names = TRUE)

	report_file_hdr <- unlist(strsplit(report_file, split = "/"))
	report_file_hdr <- report_file_hdr[length(report_file_hdr)]

	header <- read.table(unz(report_file, sub(pattern = "zip$", replacement = "txt", report_file_hdr)), 
    	                  sep = "\t", skip = 1, nrows = 2)
	report <- fread(paste("unzip -cq", report_file),
    	             skip = 9, showProgress = TRUE, sep = "\t")
	#samples <- fread(paste0("unzip -cq ",  sample_file))
	colnames(report) <- gsub(" |-", "_", colnames(report))

	geno <- report %>%
    	select(Sample_ID, SNP_Name, Allele1___Forward, Allele2___Forward) %>%
    	mutate(Sample_ID = factor(Sample_ID, levels = unique(report$Sample_ID))) %>%
    	unite(genotype, c("Allele1___Forward", "Allele2___Forward"), sep = "") %>%
    	spread(Sample_ID, genotype)

    geno_files[[i]] <- geno

}

all_genos <- Reduce(function(x, y) merge(x, y,by=c("SNP_Name"),all=TRUE),geno_files)
rownames(all_genos) <- all_genos$SNP_Name
all_genos_edit <- all_genos[c(2:length(all_genos))]

all_genos_t <- as.data.frame(t(all_genos_edit))
all_genos_t$sample.ids <- rownames(all_genos_t)

rqtl.genos.t <- all_genos_t[all_genos_t$sample.ids %in%concord.mice,]
rqtl.genos <- all_genos_edit

############################

##Pulling in haploQA snp level data for selected samples

main.path.2 = "/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/haploQA/"
files <- dir(path = main.path.2, pattern = "_snp_report.txt$", full.names = TRUE)

haploqa_files <- list()
for (i in (1:length(files))){

	file <- read.table(files[i], header=T, sep="\t")
	#sample.id <- unique(file$original_sample_id)
	rownames(file) <- file$snp_id
	file$genotype <- paste0(file$allele1_fwd, file$allele2_fwd)
	names(file)[10] <-c(unique(as.character(file$original_sample_id)))
	file.new <- file[c(3,10)]

	haploqa_files[[i]] <- file.new

}

all_haploqa_files <- Reduce(function(x, y) merge(x, y,by=c("snp_id"),all=TRUE),haploqa_files)
rownames(all_haploqa_files) <- all_haploqa_files$snp_id
all_haploqa_files_edit <- all_haploqa_files[c(2:length(all_haploqa_files))]

haploqa.genos.t <- as.data.frame(t(all_haploqa_files_edit))
haploqa.genos.t$sample.ids <- rownames(haploqa.genos.t)
haploqa.genos <- all_haploqa_files_edit

############################

## getting out phased information for r/qtl2

##control mice phase info
#apr <- readRDS("/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/genoprobs/all.samples/Jackson_Lab_Korstanje__GigaMUGA_genoprobs_8state_sorted.rds")
#apr.new <- apr[concord.mice,]
#ma <- maxmarg(apr.new, minprob = 0.9)
##mac <- do.call(cbind,ma)
#mac <- do.call("abind",list(ma,along = 2))
#mact <- as.data.frame(t(mac))
##pha <- guess_phase(gm, ma)
##phac <- do.call("abind",list(pha,along = 2))
###do mice info
#pr <- readRDS("/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/genoprobs/all.samples/Jackson_Lab_Korstanje__GigaMUGA_genoprobs_36state_sorted.rds")
#pr.new <- pr[concord.mice,]
#mp <- maxmarg(pr.new, minprob = 0.5)
#mpc <- do.call("abind",list(mp,along = 2))
#mpct <- as.data.frame(t(mpc))
#ph <- guess_phase(gm, mp)
#phc <- do.call("abind",list(ph,along = 2))

############################

## getting out phased information for haploQA

mcols <- readRDS("/Users/corneb/Documents/MyJax/CS/Projects/CBA/Adams/Data/cc_do_founders_key.RDS")
mcols$numerical.code <- 1:nrow(mcols)

main.path.2 = "/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/haploQA"
files <- dir(path = main.path.2, pattern = "_snp_report.txt$", full.names = TRUE)

haploqa_ph_files <- list()
for (i in (1:length(files))){

    file <- read.table(files[i], header=T, sep="\t")
    #129S1/SvImJ WSB/EiJ     A/J         C57BL/6J    NOD/ShiLtJ  CAST/EiJ    PWK/PhJ     NZO/HlLtJ 
    #sample.id <- unique(file$original_sample_id)
    file$haplotype1.num <- ifelse(file$haplotype1=="A/J",1,
                           ifelse(file$haplotype1=="C57BL/6J",2,
                           ifelse(file$haplotype1=="129S1/SvImJ",3,
                           ifelse(file$haplotype1=="NOD/ShiLtJ",4,
                           ifelse(file$haplotype1=="NZO/HlLtJ",5,
                           ifelse(file$haplotype1=="CAST/EiJ",6,
                           ifelse(file$haplotype1=="PWK/PhJ",7,
                           ifelse(file$haplotype1=="WSB/EiJ",8,''))))))))
    file$haplotype2.num <- ifelse(file$haplotype2=="A/J",1,
                           ifelse(file$haplotype2=="C57BL/6J",2,
                           ifelse(file$haplotype2=="129S1/SvImJ",3,
                           ifelse(file$haplotype2=="NOD/ShiLtJ",4,
                           ifelse(file$haplotype2=="NZO/HlLtJ",5,
                           ifelse(file$haplotype2=="CAST/EiJ",6,
                           ifelse(file$haplotype2=="PWK/PhJ",7,
                           ifelse(file$haplotype2=="WSB/EiJ",8,''))))))))
    rownames(file) <- file$snp_id
    #file$genotype <- paste0(file$allele1_fwd, file$allele2_fwd)
    names(file)[10] <-c(paste0("haplotype1_", unique(as.character(file$original_sample_id))))
    names(file)[11] <-c(paste0("haplotype2_", unique(as.character(file$original_sample_id))))
    file$index <- 1:nrow(file)
    file.new <- file[c(3,12,10,11)]
    haploqa_ph_files[[i]] <- file.new

}

all_haploqa_ph_files <- Reduce(function(x, y) merge(x, y,by=c("snp_id","index"),all=TRUE),haploqa_ph_files)
rownames(all_haploqa_ph_files) <- all_haploqa_ph_files$snp_id
#all_haploqa_ph_files_edit <- all_haploqa_ph_files[c(2:length(all_haploqa_ph_files))]
all_haploqa_ph_files_edit <- all_haploqa_ph_files[order(all_haploqa_ph_files$index),]
all_haploqa_ph_files_edit <- all_haploqa_ph_files_edit[c(3:length(all_haploqa_ph_files_edit))]

haploqa.ph.genos.t <- as.data.frame(t(all_haploqa_ph_files_edit))
haploqa.ph.genos.t$sample.ids <- rownames(haploqa.ph.genos.t)
haploqa.ph.genos <- all_haploqa_ph_files_edit

```

### Concordance (SNP-level genotypes)
```{r concordance letters, eval=TRUE, fig.height = 6, fig.width = 9.5, fig.align = "center"}

#merging two data frames together: genos & haploqa on snp.id

all_genos.new <- all_genos[,colnames(all_genos) %in%concord.mice]
all_genos.new$SNP_Name <- rownames(all_genos.new)
geno.letters <- merge(all_genos.new,all_haploqa_files, by.x=c("SNP_Name"), by.y=c("snp_id"), all=T)

concord.mice.2 = set_names(colnames(geno.letters[2:length(geno.letters)]))

concordance_letters_hm = function(x, y) {

    #sample.match <- ifelse(geno.letters[["B6 - 1616A"]] == geno.letters[["B6 - 1616"]], 1, 0)
    sample.match <- ifelse(geno.letters[[x]] == geno.letters[[y]], 1, 0)
    concordance.id1 <- x
    concordance.id2 <- y
    concordance.sum <- sum(sample.match)
    concordance.percentage <- (concordance.sum/nrow(geno.letters))*100
    concordance.letters <- cbind(concordance.id1,concordance.id2,concordance.percentage)
    write.table(concordance.letters,paste0("/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/concordance/concordance.letters_",x,"_vs_",y,".txt"), quote=F, sep="\t")

}

all.pairs <- map(concord.mice.2, ~map(concord.mice.2, concordance_letters_hm, y = .x))

main.path.3 = "/Users/corneb/Documents/MyJax/CS/Projects/Oconnor/concordance/"
files <- dir(path = main.path.3, pattern = ".txt$", full.names = TRUE)

concordance_letters <- list()
for (i in (1:length(files))){
    #sample.id <- paste0("^",i,"[.]")
    file <- read.table(files[i], header=T, sep="\t")
    concordance_letters[[i]] <- file
}

concordance <- as.data.frame(do.call(rbind, concordance_letters))

#removing those lines where comparing with same platform
concordance.subset.1 <- subset(concordance,  sapply(strsplit(concordance$concordance.id1,'.',fixed=T),function(x) x[2]) != sapply(strsplit(concordance$concordance.id2,'.',fixed=T),function(x) x[2]))
concordance.subset.2 <- subset(concordance.subset.1,  sapply(strsplit(concordance.subset.1$concordance.id1,'.',fixed=T),function(x) x[2]) != 'y')
concordance.subset.2 <- as.data.frame(concordance.subset.2)

#Removing .x and .y
concordance.subset.2$concordance.id1 <- gsub(".x", "", concordance.subset.2$concordance.id1)
concordance.subset.2$concordance.id2 <- gsub(".y", "", concordance.subset.2$concordance.id2)

#Rmoving same pairs

#names(concordance) <- c("concordance [snp-level genotypes]")
#concordance$Neogen_Sample_ID <- rownames(concordance)
#concordance$Neogen_Sample_ID_2 <- rownames(concordance)

##Image of all sample IDs compared to each other (from two different paltforms)
image = (p <- ggplot(concordance.subset.2, aes(concordance.id1, concordance.id2)) + 
geom_tile(aes(fill = concordance.percentage)) + 
#geom_text(aes(fill = concordance.percentage, label = concordance.percentage), size=0.1) +
scale_fill_gradient2(low = "blue", high = "red") +
xlab("R/qtl2 Data") +
ylab("haploQA Data") +
ggtitle("Concordance between samples [R/qtl2 vs. haploQA data]") + 
theme(axis.ticks = element_line(size = 0.01),
      axis.text.x =  element_text(angle=90),
      axis.title.x = element_text(vjust=-1),
      axis.title.y = element_text(vjust=2.3),
      #axis.text.y =  element_text(size=8),
      plot.title = element_text(vjust=2, hjust=0.4),
      legend.position = "right",
      plot.margin = unit(c(1,1,1,1), "cm")))

image

##Creating a list of unique pairs
#concordance.pairs <- subset(concordance.subset.2,  concordance.subset.2$concordance.id1 == concordance.subset.2$concordance.id2)
concordance.pairs <- concordance.subset.2
concordance.pairs$"concordance.percentage" <- paste0(round(as.numeric(concordance.pairs$"concordance.percentage"), digits=2),"%")
names(concordance.pairs) <- c("R/qtl2 Data","haploQA Data","Concordance Percentage [R/qtl2 vs haploQA]")

rownames(concordance.pairs) <- NULL

concordance.pairs[] %>%
  kable(escape = F,align = c("ccccccccccccccccc"),linesep ="\\hline") %>%
  kable_styling("striped", full_width = T) %>%
  column_spec(1:2, width = "5cms") 


##Image of all sample IDs compared to each other (R/qtl2)
concordance.subset.1 <- subset(concordance,  sapply(strsplit(concordance$concordance.id1,'.',fixed=T),function(x) x[2]) == sapply(strsplit(concordance$concordance.id2,'.',fixed=T),function(x) x[2]))
concordance.subset.2 <- subset(concordance.subset.1,  sapply(strsplit(concordance.subset.1$concordance.id1,'.',fixed=T),function(x) x[2]) != 'y')
concordance.subset.2 <- as.data.frame(concordance.subset.2)

#Removing .x and .y
concordance.subset.2$concordance.id1 <- gsub(".x", "", concordance.subset.2$concordance.id1)
concordance.subset.2$concordance.id2 <- gsub(".x", "", concordance.subset.2$concordance.id2)

image = (p <- ggplot(concordance.subset.2, aes(concordance.id1, concordance.id2)) + 
geom_tile(aes(fill = concordance.percentage)) + 
#geom_text(aes(fill = concordance.percentage, label = concordance.percentage), size=0.1) +
scale_fill_gradient2(low = "blue", high = "red") +
xlab("Individual 1") +
ylab("Individual 2") +
ggtitle("Concordance between samples [R/qtl2 data]") + 
theme(axis.ticks = element_line(size = 0.01),
      axis.text.x =  element_text(angle=90),
      axis.title.x = element_text(vjust=-1),
      axis.title.y = element_text(vjust=2.3),
      #axis.text.y =  element_text(size=8),
      plot.title = element_text(vjust=2, hjust=0.5),
      legend.position = "right",
      plot.margin = unit(c(1,1,1,1), "cm")))

image

##Creating a list of unique pairs
#concordance.pairs <- subset(concordance.subset.2,  concordance.subset.2$concordance.id1 == concordance.subset.2$concordance.id2)
#concordance.pairs <- concordance.subset.2
concordance.pairs <- concordance.subset.2[!duplicated(t(apply(concordance.subset.2, 1, sort))),]
concordance.pairs$"concordance.percentage" <- paste0(round(as.numeric(concordance.pairs$"concordance.percentage"), digits=2),"%")
names(concordance.pairs) <- c("Individual 1","Individual 2","Concordance Percentage [R/qtl2]")

rownames(concordance.pairs) <- NULL

concordance.pairs[] %>%
  kable(escape = F,align = c("ccccccccccccccccc"),linesep ="\\hline") %>%
  kable_styling("striped", full_width = T) %>%
  column_spec(1:2, width = "5cms") 

##Image of all sample IDs compared to each other (haploQA)
concordance.subset.2 <- subset(concordance.subset.1,  sapply(strsplit(concordance.subset.1$concordance.id1,'.',fixed=T),function(x) x[2]) != 'x')
concordance.subset.2 <- as.data.frame(concordance.subset.2)

#Removing .x and .y
concordance.subset.2$concordance.id1 <- gsub(".y", "", concordance.subset.2$concordance.id1)
concordance.subset.2$concordance.id2 <- gsub(".y", "", concordance.subset.2$concordance.id2)

image = (p <- ggplot(concordance.subset.2, aes(concordance.id1, concordance.id2)) + 
geom_tile(aes(fill = concordance.percentage)) + 
#geom_text(aes(fill = concordance.percentage, label = concordance.percentage), size=0.1) +
scale_fill_gradient2(low = "blue", high = "red") +
xlab("Individual 1") +
ylab("Individual 2") +
ggtitle("Concordance between samples [haploQA data]") + 
theme(axis.ticks = element_line(size = 0.01),
      axis.text.x =  element_text(angle=90),
      axis.title.x = element_text(vjust=-1),
      axis.title.y = element_text(vjust=2.3),
      #axis.text.y =  element_text(size=8),
      plot.title = element_text(vjust=2, hjust=0.5),
      legend.position = "right",
      plot.margin = unit(c(1,1,1,1), "cm")))

image

##Creating a list of unique pairs
#concordance.pairs <- subset(concordance.subset.2,  concordance.subset.2$concordance.id1 == concordance.subset.2$concordance.id2)
#concordance.pairs <- concordance.subset.2
concordance.pairs <- concordance.subset.2[!duplicated(t(apply(concordance.subset.2, 1, sort))),]
concordance.pairs$"concordance.percentage" <- paste0(round(as.numeric(concordance.pairs$"concordance.percentage"), digits=2),"%")
names(concordance.pairs) <- c("Individual 1","Individual 2","Concordance Percentage [haploQA]")

rownames(concordance.pairs) <- NULL

concordance.pairs[] %>%
  kable(escape = F,align = c("ccccccccccccccccc"),linesep ="\\hline") %>%
  kable_styling("striped", full_width = T) %>%
  column_spec(1:2, width = "5cms") 

```


```{r concordance numbers, eval=TRUE, include=FALSE, fig.height = 6, fig.width = 9.5, fig.align = "center"}
### Concordance between R/qtl2 & haploQA (phased haplotypes genotypes) 

#concordance.phase_list <- list()
#for(i in rownames(phc)){

#    i = "5257"
#    ind.rqtl <- as.data.frame(phc[i,,])
#    ind.rqtl$snp.id <- rownames(ind.rqtl)
#    ind.rqtl[is.na(ind.rqtl)] <- ''
#    ind.hpqa <- haploqa.ph.genos[,grep(paste0("_",i,"$"),colnames(haploqa.ph.genos))]
#    ind.hpqa$snp.id <- rownames(ind.hpqa)
#    ind.hpqa[is.na(ind.hpqa)] <- ''
#    ind <- merge(ind.rqtl,ind.hpqa, by=c("snp.id"), all=T)
#    ind[is.na(ind)] <- ''
#    ind$hap1.match <- ifelse(ind[c(2)] == ind[c(4)], 1, 0)
#    ind$hap2.match <- ifelse(ind[c(3)] == ind[c(5)], 1, 0)
#    concordance.sum.hap1 <- sum(ind$hap1.match)
#    concordance.sum.hap2 <- sum(ind$hap2.match)
#    concordance.percentage.hap1 <- (concordance.sum.hap1/nrow(ind))*100
#    concordance.percentage.hap2 <- (concordance.sum.hap2/nrow(ind))*100
#    concordance.phase <- cbind(i,concordance.percentage.hap1,concordance.percentage.hap2)
#    concordance.phase_list[[i]] <- concordance.phase

#}

#concordance.ph <- as.data.frame(do.call(rbind, concordance.phase_list))
#concordance.ph$concordance.percentage.hap1 <- paste0(round(as.numeric(concordance.ph$concordance.percentage.hap1), digits=2),"%")
#concordance.ph$concordance.percentage.hap2 <- paste0(round(as.numeric(concordance.ph$concordance.percentage.hap2), digits=2),"%")
#names(concordance.ph) <- c("mouse id", "concordance haplotype 1", "concordance haplotype 2")

#rownames(concordance.ph) <- NULL

#concordance.ph[] %>%
#  kable(escape = F,align = c("ccccccccccccccccc"),linesep ="\\hline") %>%
#  kable_styling("striped", full_width = T) %>%
#  column_spec(1:2, width = "5cms") 

```